diff -urN gcc-4.5.3.orig/libjava/classpath/configure gcc-4.5.3/libjava/classpath/configure
--- gcc-4.5.3.orig/libjava/classpath/configure	2010-06-03 10:13:04.000000000 -0700
+++ gcc-4.5.3/libjava/classpath/configure	2012-09-23 04:28:01.747902400 -0700
@@ -3157,7 +3157,7 @@
 	cp_module=""
 	;;
 	*)
-	cp_module="-module"
+	cp_module="-module -avoid-version"
 	;;
 esac
 
diff -urN gcc-4.5.3.orig/libjava/classpath/configure.ac gcc-4.5.3/libjava/classpath/configure.ac
--- gcc-4.5.3.orig/libjava/classpath/configure.ac	2010-06-03 10:13:04.000000000 -0700
+++ gcc-4.5.3/libjava/classpath/configure.ac	2012-09-23 04:27:14.830131200 -0700
@@ -63,7 +63,7 @@
 	cp_module=""
 	;;
 	*)
-	cp_module="-module"
+	cp_module="-module -avoid-version"
 	;;
 esac
 
