diff -urN automake-1.11.1.orig/INSTALL automake-1.11.1/INSTALL
--- automake-1.11.1.orig/INSTALL	2009-12-08 10:17:16.000000000 -0800
+++ automake-1.11.1/INSTALL	2012-04-03 01:07:36.010768800 -0700
@@ -142,10 +142,11 @@
 ==================
 
    By default, `make install' installs the package's commands under
-`/usr/local/bin', include files under `/usr/local/include', etc.  You
-can specify an installation prefix other than `/usr/local' by giving
-`configure' the option `--prefix=PREFIX', where PREFIX must be an
-absolute file name.
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/bin', include files under
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/include', etc.  You can specify
+an installation prefix other than `@GENTOO_PORTAGE_EPREFIX@/usr/local'
+by giving `configure' the option `--prefix=PREFIX', where PREFIX must be
+an absolute file name.
 
    You can specify separate installation prefixes for
 architecture-specific files and architecture-independent files.  If you
@@ -306,7 +307,9 @@
 Unfortunately, this technique does not work for `CONFIG_SHELL' due to
 an Autoconf bug.  Until the bug is fixed you can use this workaround:
 
-     CONFIG_SHELL=/bin/bash /bin/bash ./configure CONFIG_SHELL=/bin/bash
+     CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash \
+     	@GENTOO_PORTAGE_EPREFIX@/bin/bash ./configure \
+	CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 `configure' Invocation
 ======================
diff -urN automake-1.11.1.orig/TODO automake-1.11.1/TODO
--- automake-1.11.1.orig/TODO	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/TODO	2012-04-03 01:54:52.305414300 -0700
@@ -449,7 +449,7 @@
 must document the targets required for integration with
 non-automake-using subdirs
 
-document the "make SHELL='/bin/sh -x'" trick for debugging
+document the "make SHELL='@GENTOO_PORTAGE_EPREFIX@/bin/sh -x'" trick for debugging
 
 section on relationship to GNU make.  include notes on parallel makes
 
diff -urN automake-1.11.1.orig/bootstrap automake-1.11.1/bootstrap
--- automake-1.11.1.orig/bootstrap	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/bootstrap	2012-04-03 00:10:09.658136100 -0700
@@ -20,7 +20,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # Find the path separator
-echo "#! /bin/sh" >boot$$.sh
+echo "#! @GENTOO_PORTAGE_EPREFIX@/bin/sh" >boot$$.sh
 echo  "exit 0"   >>boot$$.sh
 chmod +x boot$$.sh
 if (PATH="/nonexistent;."; boot$$.sh) >/dev/null 2>&1; then
@@ -64,7 +64,7 @@
 if test -n "$DJDIR"; then
     BOOTSTRAP_SHELL=/dev/env/DJDIR/bin/bash.exe
 else
-    BOOTSTRAP_SHELL=/bin/sh
+    BOOTSTRAP_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/sh
 fi
 
 # Read the rule for calculating APIVERSION and execute it
diff -urN automake-1.11.1.orig/configure automake-1.11.1/configure
--- automake-1.11.1.orig/configure	2009-12-08 10:15:46.000000000 -0800
+++ automake-1.11.1/configure	2012-04-03 01:59:22.410424500 -0700
@@ -76,8 +76,8 @@
 # The user is always right.
 if test "${PATH_SEPARATOR+set}" != set; then
   PATH_SEPARATOR=:
-  (PATH='/bin;/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
-    (PATH='/bin:/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
+  (PATH='@GENTOO_PORTAGE_EPREFIX@/bin;@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
+    (PATH='@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
       PATH_SEPARATOR=';'
   }
 fi
@@ -184,7 +184,7 @@
 else
   as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
 as_found=false
-for as_dir in /bin$PATH_SEPARATOR/usr/bin$PATH_SEPARATOR$PATH
+for as_dir in @GENTOO_PORTAGE_EPREFIX@/bin$PATH_SEPARATOR@GENTOO_PORTAGE_EPREFIX@/usr/bin$PATH_SEPARATOR$PATH
 do
   IFS=$as_save_IFS
   test -z "$as_dir" && as_dir=.
@@ -240,7 +240,7 @@
 fi
 fi
 fi
-SHELL=${CONFIG_SHELL-/bin/sh}
+SHELL=${CONFIG_SHELL-@GENTOO_PORTAGE_EPREFIX@/bin/sh}
 export SHELL
 # Unset more variables known to interfere with behavior of common tools.
 CLICOLOR_FORCE= GREP_OPTIONS=
@@ -540,7 +540,7 @@
 #
 # Initializations.
 #
-ac_default_prefix=/usr/local
+ac_default_prefix=@GENTOO_PORTAGE_EPREFIX@/usr/local
 ac_clean_files=
 ac_config_libobj_dir=.
 LIBOBJS=
@@ -685,7 +685,7 @@
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
-oldincludedir='/usr/include'
+oldincludedir='@GENTOO_PORTAGE_EPREFIX@/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
 infodir='${datarootdir}/info'
 htmldir='${docdir}'
@@ -1229,7 +1229,7 @@
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
-  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
+  --oldincludedir=DIR     C header files for non-gcc [@GENTOO_PORTAGE_EPREFIX@/usr/include]
   --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
   --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
   --infodir=DIR           info documentation [DATAROOTDIR/info]
@@ -1361,8 +1361,8 @@
 uname -s = `(uname -s) 2>/dev/null || echo unknown`
 uname -v = `(uname -v) 2>/dev/null || echo unknown`
 
-/usr/bin/uname -p = `(/usr/bin/uname -p) 2>/dev/null || echo unknown`
-/bin/uname -X     = `(/bin/uname -X) 2>/dev/null     || echo unknown`
+@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p = `(@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p) 2>/dev/null || echo unknown`
+@GENTOO_PORTAGE_EPREFIX@/bin/uname -X     = `(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X) 2>/dev/null     || echo unknown`
 
 /bin/arch              = `(/bin/arch) 2>/dev/null              || echo unknown`
 /usr/bin/arch -k       = `(/usr/bin/arch -k) 2>/dev/null       || echo unknown`
@@ -1792,7 +1792,7 @@
     # Account for people who put trailing slashes in PATH elements.
 case $as_dir/ in #((
   ./ | .// | /[cC]/* | \
-  /etc/* | /usr/sbin/* | /usr/etc/* | /sbin/* | /usr/afsws/bin/* | \
+  @GENTOO_PORTAGE_EPREFIX@/etc/* | @GENTOO_PORTAGE_EPREFIX@/usr/sbin/* | @GENTOO_PORTAGE_EPREFIX@/usr/etc/* | @GENTOO_PORTAGE_EPREFIX@/sbin/* | /usr/afsws/bin/* | \
   ?:[\\/]os2[\\/]install[\\/]* | ?:[\\/]OS2[\\/]INSTALL[\\/]* | \
   /usr/ucb/* ) ;;
   *)
@@ -2146,7 +2146,7 @@
   $as_echo_n "(cached) " >&6
 else
   cat >conftest.make <<\_ACEOF
-SHELL = /bin/sh
+SHELL = @GENTOO_PORTAGE_EPREFIX@/bin/sh
 all:
 	@echo '@@@%%%=$(MAKE)=@@@%%%'
 _ACEOF
@@ -2701,12 +2701,12 @@
 
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether /bin/sh has working 'set -e' with exit trap" >&5
-$as_echo_n "checking whether /bin/sh has working 'set -e' with exit trap... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether @GENTOO_PORTAGE_EPREFIX@/bin/sh has working 'set -e' with exit trap" >&5
+$as_echo_n "checking whether @GENTOO_PORTAGE_EPREFIX@/bin/sh has working 'set -e' with exit trap... " >&6; }
 if test "${am_cv_sh_errexit_works+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
-  if /bin/sh -ec "trap 'exit \$?' 0; (exit 77); exit 77"; test $? = 77
+  if @GENTOO_PORTAGE_EPREFIX@/bin/sh -ec "trap 'exit \$?' 0; (exit 77); exit 77"; test $? = 77
 then
   am_cv_sh_errexit_works=yes
 else
@@ -2955,8 +2955,8 @@
 # The user is always right.
 if test "${PATH_SEPARATOR+set}" != set; then
   PATH_SEPARATOR=:
-  (PATH='/bin;/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
-    (PATH='/bin:/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
+  (PATH='@GENTOO_PORTAGE_EPREFIX@/bin;@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
+    (PATH='@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
       PATH_SEPARATOR=';'
   }
 fi
diff -urN automake-1.11.1.orig/configure.ac automake-1.11.1/configure.ac
--- automake-1.11.1.orig/configure.ac	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/configure.ac	2012-04-03 00:14:09.450585000 -0700
@@ -171,9 +171,9 @@
 AC_PROG_EGREP
 AC_PROG_FGREP
 
-AC_CACHE_CHECK([whether /bin/sh has working 'set -e' with exit trap],
+AC_CACHE_CHECK([whether @GENTOO_PORTAGE_EPREFIX@/bin/sh has working 'set -e' with exit trap],
 [am_cv_sh_errexit_works],
-[if /bin/sh -ec "trap 'exit \$?' 0; (exit 77); exit 77"; test $? = 77
+[if @GENTOO_PORTAGE_EPREFIX@/bin/sh -ec "trap 'exit \$?' 0; (exit 77); exit 77"; test $? = 77
 then
   am_cv_sh_errexit_works=yes
 else
diff -urN automake-1.11.1.orig/doc/automake.info-1 automake-1.11.1/doc/automake.info-1
--- automake-1.11.1.orig/doc/automake.info-1	2009-12-08 10:15:55.000000000 -0800
+++ automake-1.11.1/doc/automake.info-1	2012-04-03 00:28:05.044170300 -0700
@@ -457,10 +457,10 @@
 you should find a copy of this file in
 `PREFIX/share/doc/automake/amhello-1.0.tar.gz', where PREFIX is the
 installation prefix specified during configuration (PREFIX defaults to
-`/usr/local', however if Automake was installed by some GNU/Linux
-distribution it most likely has been set to `/usr').  If you do not
-have a copy of Automake installed, you can find a copy of this file
-inside the `doc/' directory of the Automake package.
+`@GENTOO_PORTAGE_EPREFIX@/usr/local', however if Automake was installed
+by some GNU/Linux distribution it most likely has been set to `/usr').
+If you do not have a copy of Automake installed, you can find a copy of
+this file inside the `doc/' directory of the Automake package.
 
    Some of the following use cases present features that are in fact
 extensions to the GNU Build System.  Read: they are not specified by
@@ -541,12 +541,14 @@
 header files, scripts, and other data files from the source directory
 to their final destination on the system.  The command `make install'
 will do that.  However, by default everything will be installed in
-subdirectories of `/usr/local': binaries will go into `/usr/local/bin',
-libraries will end up in `/usr/local/lib', etc.  This destination is
+subdirectories of `@GENTOO_PORTAGE_EPREFIX@/usr/local': binaries will
+go into `@GENTOO_PORTAGE_EPREFIX@/usr/local/bin', libraries will end
+up in `@GENTOO_PORTAGE_EPREFIX@/usr/local/lib', etc.  This destination is
 usually not writable by any user, so we assume that we have to become
 root before we can run `make install'.  In our example, running `make
-install' will copy the program `hello' into `/usr/local/bin' and
-`README' into `/usr/local/share/doc/amhello'.
+install' will copy the program `hello' into
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/bin' and
+`README' into `@GENTOO_PORTAGE_EPREFIX@/usr/local/share/doc/amhello'.
 
    A last and optional step is to run `make installcheck'.  This
 command may run tests on the installed files.  `make check' tests the
@@ -621,7 +623,7 @@
 
 Directory variable   Default value
 ------------------------------------------------------- 
-`prefix'             `/usr/local'
+`prefix'             `@GENTOO_PORTAGE_EPREFIX@/usr/local'
   `exec_prefix'        `${prefix}'
     `bindir'             `${exec_prefix}/bin'
     `libdir'             `${exec_prefix}/lib'
@@ -638,10 +640,10 @@
 name.  In a package, any installable file will be installed in one of
 these directories.  For instance in `amhello-1.0', the program `hello'
 is to be installed in BINDIR, the directory for binaries.  The default
-value for this directory is `/usr/local/bin', but the user can supply a
-different value when calling `configure'.  Also the file `README' will
-be installed into DOCDIR, which defaults to
-`/usr/local/share/doc/amhello'.
+value for this directory is `@GENTOO_PORTAGE_EPREFIX@/usr/local/bin',
+but the user can supply a different value when calling `configure'.
+Also the file `README' will be installed into DOCDIR, which defaults to
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/share/doc/amhello'.
 
    A user who wishes to install a package on his own account could
 proceed as follows:
@@ -921,7 +923,7 @@
 GNU/Linux PC.
 
      ~/amhello-1.0 % ./configure --build i686-pc-linux-gnu --host i586-mingw32msvc
-     checking for a BSD-compatible install... /usr/bin/install -c
+     checking for a BSD-compatible install... @GENTOO_PORTAGE_EPREFIX@/usr/bin/install -c
      checking whether build environment is sane... yes
      checking for gawk... gawk
      checking whether make sets $(MAKE)... yes
@@ -984,7 +986,7 @@
      Run `sed PROGRAM' on installed program names.
 
    The following commands would install `hello' as
-`/usr/local/bin/test-hello', for instance.
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/bin/test-hello', for instance.
 
      ~/amhello-1.0 % ./configure --program-prefix test-
      ...
@@ -1297,7 +1299,7 @@
 names `config.h', `Makefile', and `src/Makefile'.  Let's do this:
 
      ~/amhello % ./configure
-     checking for a BSD-compatible install... /usr/bin/install -c
+     checking for a BSD-compatible install... @GENTOO_PORTAGE_EPREFIX@/usr/bin/install -c
      checking whether build environment is sane... yes
      checking for gawk... no
      checking for mawk... mawk
@@ -2857,28 +2859,28 @@
 `automake' version `1.6.2' is installed with `--prefix=/usr' by the
 system vendor.  Thus, the default search directories are
 
-  1. `/usr/share/aclocal-1.6/'
+  1. `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal-1.6/'
 
-  2. `/usr/share/aclocal/'
+  2. `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/'
 
    However, suppose further that many packages have been manually
 installed on the system, with $prefix=/usr/local, as is typical.  In
 that case, many of these "extra" `.m4' files are in
-`/usr/local/share/aclocal'.  The only way to force `/usr/bin/aclocal'
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal'.  The only way to force `@GENTOO_PORTAGE_EPREFIX@/usr/bin/aclocal'
 to find these "extra" `.m4' files is to always call `aclocal -I
-/usr/local/share/aclocal'.  This is inconvenient.  With `dirlist', one
-may create a file `/usr/share/aclocal/dirlist' containing only the
+@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal'.  This is inconvenient.  With `dirlist', one
+may create a file `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/dirlist' containing only the
 single line
 
-     /usr/local/share/aclocal
+     @GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal
 
    Now, the "default" search path on the affected system is
 
-  1. `/usr/share/aclocal-1.6/'
+  1. `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal-1.6/'
 
-  2. `/usr/share/aclocal/'
+  2. `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/'
 
-  3. `/usr/local/share/aclocal/'
+  3. `@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal/'
 
    without the need for `-I' options; `-I' options can be reserved for
 project-specific needs (`my-source-dir/m4/'), rather than using it to
@@ -3095,8 +3097,8 @@
    Here is a use case that illustrates the use of `--install' and its
 interaction with serial numbers.  Let's assume we maintain a package
 called MyPackage, the `configure.ac' of which requires a third-party
-macro `AX_THIRD_PARTY' defined in `/usr/share/aclocal/thirdparty.m4' as
-follows:
+macro `AX_THIRD_PARTY' defined in
+`@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4' as follows:
 
      # serial 1
      AC_DEFUN([AX_THIRD_PARTY], [...])
@@ -3117,13 +3119,13 @@
 
    * No local macros define `AX_THIRD_PARTY'
 
-   * `/usr/share/aclocal/thirdparty.m4' defines `AX_THIRD_PARTY' with
-     serial 1.
+   * `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4' defines
+     `AX_THIRD_PARTY' with serial 1.
 
-Because `/usr/share/aclocal/thirdparty.m4' is a system-wide macro and
-`aclocal' was given the `--install' option, it will copy this file in
-`m4/thirdparty.m4', and output an `aclocal.m4' that contains
-`m4_include([m4/thirdparty.m4])'.
+Because `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4' is a
+system-wide macro and `aclocal' was given the `--install' option, it will
+copy this file in `m4/thirdparty.m4', and output an `aclocal.m4' that
+contains `m4_include([m4/thirdparty.m4])'.
 
    The next time `aclocal -I m4 --install' is run (either via
 `autoreconf', by hand, or from the `Makefile' rebuild rules) something
@@ -3133,13 +3135,13 @@
 
    * `m4/thirdparty.m4' defines `AX_THIRD_PARTY' with serial 1.
 
-   * `/usr/share/aclocal/thirdparty.m4' defines `AX_THIRD_PARTY' with
-     serial 1.
+   * `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4' defines
+     `AX_THIRD_PARTY' with serial 1.
 
 Because both files have the same serial number, `aclocal' uses the
 first it found in its search path order (*note Macro Search Path::).
-`aclocal' therefore ignores `/usr/share/aclocal/thirdparty.m4' and
-outputs an `aclocal.m4' that contains `m4_include([m4/thirdparty.m4])'.
+`aclocal' therefore ignores `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4'
+and outputs an `aclocal.m4' that contains `m4_include([m4/thirdparty.m4])'.
 
    Local directories specified with `-I' are always searched before
 system-wide directories, so a local file will always be preferred to
@@ -3154,18 +3156,18 @@
 
    * `m4/thirdparty.m4' defines `AX_THIRD_PARTY' with serial 1.
 
-   * `/usr/share/aclocal/thirdparty.m4' defines `AX_THIRD_PARTY' with
-     serial 2.
+   * `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4'
+     defines `AX_THIRD_PARTY' with serial 2.
 
 When `aclocal' sees a greater serial number, it immediately forgets
 anything it knows from files that have the same basename and a smaller
 serial number.  So after it has found
-`/usr/share/aclocal/thirdparty.m4' with serial 2, `aclocal' will
-proceed as if it had never seen `m4/thirdparty.m4'.  This brings us
-back to a situation similar to that at the beginning of our example,
-where no local file defined the macro.  `aclocal' will install the new
-version of the macro in `m4/thirdparty.m4', in this case overriding the
-old version.  MyPackage just had its macro updated as a side effect of
+`@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4' with serial 2,
+`aclocal' will proceed as if it had never seen `m4/thirdparty.m4'.  This
+brings us back to a situation similar to that at the beginning of our
+example, where no local file defined the macro.  `aclocal' will install
+the new version of the macro in `m4/thirdparty.m4', in this case overriding
+the old version.  MyPackage just had its macro updated as a side effect of
 running `aclocal'.
 
    If you are leery of letting `aclocal' update your local macro, you
@@ -3219,11 +3221,11 @@
 directly anymore.
 
    For the time being, third-party packages should continue to install
-public macros into `/usr/share/aclocal/'.  If `aclocal' is replaced by
-another tool it might make sense to rename the directory, but
-supporting `/usr/share/aclocal/' for backward compatibility should be
-really easy provided all macros are properly written (*note Extending
-aclocal::).
+public macros into `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/'.  If
+`aclocal' is replaced by another tool it might make sense to rename the
+directory, but supporting `@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/'
+for backward compatibility should be really easy provided all macros
+are properly written (*note Extending aclocal::).
 
 
 File: automake.info,  Node: Macros,  Prev: Invoking aclocal,  Up: configure
@@ -6192,7 +6194,7 @@
      make[1]: Entering directory `/home/adl/tmp'
      source='foo.c' object='foo.o' libtool=no \
      depfile='.deps/foo.Po' tmpdepfile='.deps/foo.TPo' \
-     depmode=gcc /bin/sh ./depcomp \
+     depmode=gcc @GENTOO_PORTAGE_EPREFIX@/bin/sh ./depcomp \
      gcc -I. -I. -g -O2 -c `test -f 'foo.c' || echo './'`foo.c
      gcc  -g -O2   -o foo  foo.o
      make[1]: Leaving directory `/home/adl/tmp'
@@ -6209,7 +6211,7 @@
      % make foo
      source='foo.c' object='foo.o' libtool=no \
      depfile='.deps/foo.Po' tmpdepfile='.deps/foo.TPo' \
-     depmode=gcc /bin/sh ./depcomp \
+     depmode=gcc @GENTOO_PORTAGE_EPREFIX@/bin/sh ./depcomp \
      gcc -I. -I. -g -O2 -c `test -f 'foo.c' || echo './'`foo.c
      foo.c:2: bindir.h: No such file or directory
      make: *** [foo.o] Error 1
@@ -6893,15 +6895,18 @@
 with the value of `DESTDIR' before being copied into the install area.
 Here is an example of typical DESTDIR usage:
 
-     mkdir /tmp/staging &&
-     make DESTDIR=/tmp/staging install
+     mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/staging &&
+     make DESTDIR=@GENTOO_PORTAGE_EPREFIX@/tmp/staging install
 
    The `mkdir' command avoids a security problem if the attacker
-creates a symbolic link from `/tmp/staging' to a victim area; then
-`make' places install objects in a directory tree built under
-`/tmp/staging'.  If `/gnu/bin/foo' and `/gnu/share/aclocal/foo.m4' are
+creates a symbolic link from `@GENTOO_PORTAGE_EPREFIX@/tmp/staging' to
+a victim area; then `make' places install objects in a directory tree
+built under `@GENTOO_PORTAGE_EPREFIX@/tmp/staging'.  If
+`@GENTOO_PORTAGE_EPREFIX@/gnu/bin/foo' and
+`@GENTOO_PORTAGE_EPREFIX@/gnu/share/aclocal/foo.m4' are
 to be installed, the above command would install
-`/tmp/staging/gnu/bin/foo' and `/tmp/staging/gnu/share/aclocal/foo.m4'.
+`@GENTOO_PORTAGE_EPREFIX@/tmp/staging/gnu/bin/foo' and
+`@GENTOO_PORTAGE_EPREFIX@/tmp/staging/gnu/share/aclocal/foo.m4'.
 
    This feature is commonly used to build install images and packages
 (*note DESTDIR::).
diff -urN automake-1.11.1.orig/doc/automake.info-2 automake-1.11.1/doc/automake.info-2
--- automake-1.11.1.orig/doc/automake.info-2	2009-12-08 10:15:55.000000000 -0800
+++ automake-1.11.1/doc/automake.info-2	2012-04-03 00:34:26.462438700 -0700
@@ -2455,19 +2455,20 @@
 etc.), the effect will be the same: user-installations are impossible.
 
    When a (non-root) user wants to install a package, he usually has no
-right to install anything in `/usr' or `/usr/local'.  So he does
+right to install anything in `@GENTOO_PORTAGE_EPREFIX@/usr' or
+`@GENTOO_PORTAGE_EPREFIX@/usr/local'.  So he does
 something like `./configure --prefix ~/usr' to install package in his
 own `~/usr' tree.
 
    If a package attempts to install something to some hard-coded path
-(e.g., `/etc/afile'), regardless of this `--prefix' setting, then the
-installation will fail.  `make distcheck' performs such a `--prefix'
-installation, hence it will fail too.
+(e.g., `@GENTOO_PORTAGE_EPREFIX@/etc/afile'), regardless of this
+`--prefix' setting, then the installation will fail.  `make distcheck'
+performs such a `--prefix' installation, hence it will fail too.
 
    Now, there are some easy solutions.
 
-   The above `install-data-local' example for installing `/etc/afile'
-would be better replaced by
+   The above `install-data-local' example for installing
+`@GENTOO_PORTAGE_EPREFIX@/etc/afile' would be better replaced by
 
      sysconf_DATA = afile
 
@@ -2486,7 +2487,7 @@
 
      % python -c 'from distutils import sysconfig;
                   print sysconfig.get_python_lib(1,0)'
-     /usr/lib/python2.5/site-packages
+     @GENTOO_PORTAGE_EPREFIX@/usr/lib/python2.5/site-packages
 
    If you indeed use this absolute path to install your shared library,
 non-root users will not be able to install the package, hence distcheck
diff -urN automake-1.11.1.orig/doc/automake.texi automake-1.11.1/doc/automake.texi
--- automake-1.11.1.orig/doc/automake.texi	2012-04-03 00:03:58.809481200 -0700
+++ automake-1.11.1/doc/automake.texi	2012-04-03 01:05:34.149901500 -0700
@@ -521,11 +521,11 @@
 system, you should find a copy of this file in
 @file{@var{prefix}/share/doc/automake/amhello-1.0.tar.gz}, where
 @var{prefix} is the installation prefix specified during configuration
-(@var{prefix} defaults to @file{/usr/local}, however if Automake was
-installed by some GNU/Linux distribution it most likely has been set
-to @file{/usr}).  If you do not have a copy of Automake installed,
-you can find a copy of this file inside the @file{doc/} directory of
-the Automake package.
+(@var{prefix} defaults to @file{@GENTOO_PORTAGE_EPREFIX@/usr/local},
+however if Automake was installed by some GNU/Linux distribution it
+most likely has been set to @file{/usr}).  If you do not have a copy
+of Automake installed, you can find a copy of this file inside the
+@file{doc/} directory of the Automake package.
 
 Some of the following use cases present features that are in fact
 extensions to the GNU Build System.  Read: they are not specified by
@@ -611,13 +611,18 @@
 libraries, header files, scripts, and other data files from the
 source directory to their final destination on the system.  The
 command @code{make install} will do that.  However, by default
-everything will be installed in subdirectories of @file{/usr/local}:
-binaries will go into @file{/usr/local/bin}, libraries will end up in
-@file{/usr/local/lib}, etc.  This destination is usually not writable
+everything will be installed in subdirectories of
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local}:
+binaries will go into @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin},
+libraries will end up in
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/lib}, etc.
+This destination is usually not writable
 by any user, so we assume that we have to become root before we can
 run @code{make install}.  In our example, running @code{make install}
-will copy the program @file{hello} into @file{/usr/local/bin}
-and @file{README} into @file{/usr/local/share/doc/amhello}.
+will copy the program @file{hello} into
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin}
+and @file{README} into
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/doc/amhello}.
 
 A last and optional step is to run @code{make installcheck}.  This
 command may run tests on the installed files.  @code{make check} tests
@@ -692,7 +697,7 @@
 
 @multitable {Directory variable} {@code{$@{datarootdir@}/doc/$@{PACKAGE@}}}
 @headitem Directory variable    @tab Default value
-@item @code{prefix}              @tab @code{/usr/local}
+@item @code{prefix}              @tab @code{@GENTOO_PORTAGE_EPREFIX@/usr/local}
 @item @w{@ @ @code{exec_prefix}} @tab @code{$@{prefix@}}
 @item @w{@ @ @ @ @code{bindir}}  @tab @code{$@{exec_prefix@}/bin}
 @item @w{@ @ @ @ @code{libdir}}  @tab @code{$@{exec_prefix@}/lib}
@@ -717,10 +722,11 @@
 these directories.  For instance in @code{amhello-1.0}, the program
 @file{hello} is to be installed in @var{bindir}, the directory for
 binaries.  The default value for this directory is
-@file{/usr/local/bin}, but the user can supply a different value when
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin}, but the user can supply
+a different value when
 calling @command{configure}.  Also the file @file{README} will be
 installed into @var{docdir}, which defaults to
-@file{/usr/local/share/doc/amhello}.
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/doc/amhello}.
 
 @opindex --prefix
 
@@ -913,7 +919,7 @@
 
 On the first host, you could create a local build directory:
 @example
-[HOST1] ~ % @kbd{mkdir /tmp/amh && cd /tmp/amh}
+[HOST1] ~ % @kbd{mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/amh && cd @GENTOO_PORTAGE_EPREFIX@/tmp/amh}
 [HOST1] /tmp/amh % @kbd{/nfs/src/amhello-1.0/configure}
 ...
 [HOST1] /tmp/amh % @kbd{make && sudo make install}
@@ -928,7 +934,7 @@
 On the second host, you would do exactly the same, possibly at
 the same time:
 @example
-[HOST2] ~ % @kbd{mkdir /tmp/amh && cd /tmp/amh}
+[HOST2] ~ % @kbd{mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/amh && cd @GENTOO_PORTAGE_EPREFIX@/tmp/amh}
 [HOST2] /tmp/amh % @kbd{/nfs/src/amhello-1.0/configure}
 ...
 [HOST2] /tmp/amh % @kbd{make && sudo make install}
@@ -978,25 +984,25 @@
 
 Here is how we could revisit our two-host installation example,
 assuming that (1) we want to install the package directly in
-@file{/usr}, and (2) the directory @file{/usr/share} is shared by the
-two hosts.
+@file{@GENTOO_PORTAGE_EPREFIX@/usr}, and (2) the directory
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share} is shared by the two hosts.
 
 On the first host we would run
 @example
-[HOST1] ~ % @kbd{mkdir /tmp/amh && cd /tmp/amh}
-[HOST1] /tmp/amh % @kbd{/nfs/src/amhello-1.0/configure --prefix /usr}
+[HOST1] ~ % @kbd{mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/amh && cd @GENTOO_PORTAGE_EPREFIX@/tmp/amh}
+[HOST1] @GENTOO_PORTAGE_EPREFIX@/tmp/amh % @kbd{/nfs/src/amhello-1.0/configure --prefix @GENTOO_PORTAGE_EPREFIX@/usr}
 ...
-[HOST1] /tmp/amh % @kbd{make && sudo make install}
+[HOST1] @GENTOO_PORTAGE_EPREFIX@/tmp/amh % @kbd{make && sudo make install}
 ...
 @end example
 
 On the second host, however, we need only install the
 architecture-specific files.
 @example
-[HOST2] ~ % @kbd{mkdir /tmp/amh && cd /tmp/amh}
-[HOST2] /tmp/amh % @kbd{/nfs/src/amhello-1.0/configure --prefix /usr}
+[HOST2] ~ % @kbd{mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/amh && cd @GENTOO_PORTAGE_EPREFIX@/tmp/amh}
+[HOST2] @GENTOO_PORTAGE_EPREFIX@/tmp/amh % @kbd{/nfs/src/amhello-1.0/configure --prefix @GENTOO_PORTAGE_EPREFIX@/usr}
 ...
-[HOST2] /tmp/amh % @kbd{make && sudo make install-exec}
+[HOST2] @GENTOO_PORTAGE_EPREFIX@/tmp/amh % @kbd{make && sudo make install-exec}
 ...
 @end example
 
@@ -1039,7 +1045,7 @@
 
 @smallexample
 ~/amhello-1.0 % @kbd{./configure --build i686-pc-linux-gnu --host i586-mingw32msvc}
-checking for a BSD-compatible install... /usr/bin/install -c
+checking for a BSD-compatible install... @GENTOO_PORTAGE_EPREFIX@/usr/bin/install -c
 checking whether build environment is sane... yes
 checking for gawk... gawk
 checking whether make sets $(MAKE)... yes
@@ -1111,7 +1117,7 @@
 @end table
 
 The following commands would install @file{hello}
-as @file{/usr/local/bin/test-hello}, for instance.
+as @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin/test-hello}, for instance.
 
 @example
 ~/amhello-1.0 % @kbd{./configure --program-prefix test-}
@@ -1139,7 +1145,8 @@
 
 The @code{DESTDIR} variable can be used to perform a staged
 installation.  The package should be configured as if it was going to
-be installed in its final location (e.g., @code{--prefix /usr}), but
+be installed in its final location (e.g.,
+@code{--prefix @GENTOO_PORTAGE_EPREFIX@/usr}), but
 when running @code{make install}, the @code{DESTDIR} should be set to
 the absolute name of a directory into which the installation will be
 diverted.  From this directory it is easy to review which files are
@@ -1152,7 +1159,7 @@
 snapshot of all the files to be installed.
 
 @example
-~/amhello-1.0 % @kbd{./configure --prefix /usr}
+~/amhello-1.0 % @kbd{./configure --prefix @GENTOO_PORTAGE_EPREFIX@/usr}
 @dots{}
 ~/amhello-1.0 % @kbd{make}
 @dots{}
@@ -1161,15 +1168,16 @@
 ~/amhello-1.0 % @kbd{cd ~/inst}
 ~/inst % @kbd{find . -type f -print > ../files.lst}
 ~/inst % @kbd{tar zcvf ~/amhello-1.0-i686.tar.gz `cat ../files.lst`}
-./usr/bin/hello
-./usr/share/doc/amhello/README
+.@GENTOO_PORTAGE_EPREFIX@/usr/bin/hello
+.@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/amhello/README
 @end example
 
 After this example, @code{amhello-1.0-i686.tar.gz} is ready to be
 uncompressed in @file{/} on many hosts.  (Using @code{`cat ../files.lst`}
 instead of @samp{.} as argument for @command{tar} avoids entries for
 each subdirectory in the archive: we would not like @command{tar} to
-restore the modification time of @file{/}, @file{/usr/}, etc.)
+restore the modification time of @file{@GENTOO_PORTAGE_EPREFIX@/},
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/}, etc.)
 
 Note that when building packages for several architectures, it might
 be convenient to use @code{make install-data} and @code{make
@@ -1248,7 +1256,7 @@
 can be used):
 
 @example
-~/amhello-1.0 % @kbd{./configure --prefix /usr}
+~/amhello-1.0 % @kbd{./configure --prefix @GENTOO_PORTAGE_EPREFIX@/usr}
 @dots{}
 checking dependency style of gcc... gcc3
 @dots{}
@@ -1479,7 +1487,7 @@
 
 @example
 ~/amhello % @kbd{./configure}
-checking for a BSD-compatible install... /usr/bin/install -c
+checking for a BSD-compatible install... @GENTOO_PORTAGE_EPREFIX@/usr/bin/install -c
 checking whether build environment is sane... yes
 checking for gawk... no
 checking for mawk... mawk
@@ -3223,11 +3231,11 @@
 @end table
 
 As an example, suppose that @command{automake-1.6.2} was configured with
-@option{--prefix=@-/usr/local}.  Then, the search path would be:
+@option{--prefix=@-@GENTOO_PORTAGE_EPREFIX@/usr/local}.  Then, the search path would be:
 
 @enumerate
-@item @file{/usr/local/share/aclocal-1.6/}
-@item @file{/usr/local/share/aclocal/}
+@item @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal-1.6/}
+@item @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal/}
 @end enumerate
 
 As explained in (@pxref{aclocal Options}), there are several options that
@@ -3305,35 +3313,36 @@
 
 @file{dirlist} is useful in the following situation: suppose that
 @command{automake} version @code{1.6.2} is installed with
-@samp{--prefix=/usr} by the system vendor.  Thus, the default search
-directories are
+@samp{--prefix=@GENTOO_PORTAGE_EPREFIX@/usr} by the system vendor.  Thus,
+the default search directories are
 
 @c @code looks better than @file here
 @enumerate
-@item @code{/usr/share/aclocal-1.6/}
-@item @code{/usr/share/aclocal/}
+@item @code{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal-1.6/}
+@item @code{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/}
 @end enumerate
 
 However, suppose further that many packages have been manually
 installed on the system, with $prefix=/usr/local, as is typical.  In
 that case, many of these ``extra'' @file{.m4} files are in
-@file{/usr/local/share/aclocal}.  The only way to force
-@file{/usr/bin/aclocal} to find these ``extra'' @file{.m4} files is to
-always call @samp{aclocal -I /usr/local/share/aclocal}.  This is
-inconvenient.  With @file{dirlist}, one may create a file
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal}.  The
+only way to force @file{@GENTOO_PORTAGE_EPREFIX@/usr/bin/aclocal} to
+find these ``extra'' @file{.m4} files is to always call
+@samp{aclocal -I @GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal}.
+This is inconvenient.  With @file{dirlist}, one may create a file
 @file{/usr/share/aclocal/dirlist} containing only the single line
 
 @example
-/usr/local/share/aclocal
+@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal
 @end example
 
 Now, the ``default'' search path on the affected system is
 
 @c @code looks better than @file here
 @enumerate
-@item @code{/usr/share/aclocal-1.6/}
-@item @code{/usr/share/aclocal/}
-@item @code{/usr/local/share/aclocal/}
+@item @code{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal-1.6/}
+@item @code{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/}
+@item @code{@GENTOO_PORTAGE_EPREFIX@/usr/local/share/aclocal/}
 @end enumerate
 
 without the need for @option{-I} options; @option{-I} options can be reserved
@@ -3639,7 +3648,7 @@
 @file{m4/thirdparty.m4} defines @code{AX_THIRD_PARTY}
 with serial 1.
 @item
-@file{/usr/share/aclocal/thirdparty.m4} defines @code{AX_THIRD_PARTY}
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4} defines @code{AX_THIRD_PARTY}
 with serial 1.
 @end itemize
 
@@ -3647,7 +3656,7 @@
 Because both files have the same serial number, @command{aclocal} uses
 the first it found in its search path order (@pxref{Macro Search
 Path}).  @command{aclocal} therefore ignores
-@file{/usr/share/aclocal/thirdparty.m4} and outputs an
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4} and outputs an
 @file{aclocal.m4} that contains @samp{m4_include([m4/thirdparty.m4])}.
 
 Local directories specified with @option{-I} are always searched before
@@ -3666,7 +3675,7 @@
 @file{m4/thirdparty.m4} defines @code{AX_THIRD_PARTY}
 with serial 1.
 @item
-@file{/usr/share/aclocal/thirdparty.m4} defines @code{AX_THIRD_PARTY}
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4} defines @code{AX_THIRD_PARTY}
 with serial 2.
 @end itemize
 
@@ -3674,7 +3683,7 @@
 When @command{aclocal} sees a greater serial number, it immediately
 forgets anything it knows from files that have the same basename and a
 smaller serial number.  So after it has found
-@file{/usr/share/aclocal/thirdparty.m4} with serial 2,
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/thirdparty.m4} with serial 2,
 @command{aclocal} will proceed as if it had never seen
 @file{m4/thirdparty.m4}.  This brings us back to a situation similar
 to that at the beginning of our example, where no local file defined
@@ -3736,9 +3745,10 @@
 you will not call @command{aclocal} directly anymore.
 
 For the time being, third-party packages should continue to install
-public macros into @file{/usr/share/aclocal/}.  If @command{aclocal}
-is replaced by another tool it might make sense to rename the
-directory, but supporting @file{/usr/share/aclocal/} for backward
+public macros into @file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/}.
+If @command{aclocal} is replaced by another tool it might make sense to
+rename the directory, but supporting
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/share/aclocal/} for backward
 compatibility should be really easy provided all macros are properly
 written (@pxref{Extending aclocal}).
 
@@ -7182,7 +7192,7 @@
 % make
 source='foo.c' object='foo.o' libtool=no \
 depfile='.deps/foo.Po' tmpdepfile='.deps/foo.TPo' \
-depmode=gcc /bin/sh ./depcomp \
+depmode=gcc @GENTOO_PORTAGE_EPREFIX@/bin/sh ./depcomp \
 gcc -I. -I. -g -O2 -c `test -f 'foo.c' || echo './'`foo.c
 foo.c:2: bindir.h: No such file or directory
 make: *** [foo.o] Error 1
@@ -7215,12 +7225,12 @@
 
 @example
 % make
-echo '#define bindir "/usr/local/bin"' >bindir.h
+echo '#define bindir "@GENTOO_PORTAGE_EPREFIX@/usr/local/bin"' >bindir.h
 make  all-am
 make[1]: Entering directory `/home/adl/tmp'
 source='foo.c' object='foo.o' libtool=no \
 depfile='.deps/foo.Po' tmpdepfile='.deps/foo.TPo' \
-depmode=gcc /bin/sh ./depcomp \
+depmode=gcc @GENTOO_PORTAGE_EPREFIX@/bin/sh ./depcomp \
 gcc -I. -I. -g -O2 -c `test -f 'foo.c' || echo './'`foo.c
 gcc  -g -O2   -o foo  foo.o
 make[1]: Leaving directory `/home/adl/tmp'
@@ -7239,7 +7249,7 @@
 % make foo
 source='foo.c' object='foo.o' libtool=no \
 depfile='.deps/foo.Po' tmpdepfile='.deps/foo.TPo' \
-depmode=gcc /bin/sh ./depcomp \
+depmode=gcc @GENTOO_PORTAGE_EPREFIX@/bin/sh ./depcomp \
 gcc -I. -I. -g -O2 -c `test -f 'foo.c' || echo './'`foo.c
 foo.c:2: bindir.h: No such file or directory
 make: *** [foo.o] Error 1
@@ -8069,17 +8079,17 @@
 into the install area.  Here is an example of typical DESTDIR usage:
 
 @example
-mkdir /tmp/staging &&
-make DESTDIR=/tmp/staging install
+mkdir @GENTOO_PORTAGE_EPREFIX@/tmp/staging &&
+make DESTDIR=@GENTOO_PORTAGE_EPREFIX@/tmp/staging install
 @end example
 
 The @command{mkdir} command avoids a security problem if the attacker
-creates a symbolic link from @file{/tmp/staging} to a victim area;
+creates a symbolic link from @file{@GENTOO_PORTAGE_EPREFIX@/tmp/staging} to a victim area;
 then @command{make} places install objects in a directory tree built under
-@file{/tmp/staging}.  If @file{/gnu/bin/foo} and
+@file{@GENTOO_PORTAGE_EPREFIX@/tmp/staging}.  If @file{/gnu/bin/foo} and
 @file{/gnu/share/aclocal/foo.m4} are to be installed, the above command
-would install @file{/tmp/staging/gnu/bin/foo} and
-@file{/tmp/staging/gnu/share/aclocal/foo.m4}.
+would install @file{@GENTOO_PORTAGE_EPREFIX@/tmp/staging/gnu/bin/foo} and
+@file{@GENTOO_PORTAGE_EPREFIX@/tmp/staging/gnu/share/aclocal/foo.m4}.
 
 This feature is commonly used to build install images and packages
 (@pxref{DESTDIR}).
@@ -11560,7 +11570,7 @@
 @example
 % @kbd{python -c 'from distutils import sysconfig;
              print sysconfig.get_python_lib(1,0)'}
-/usr/lib/python2.5/site-packages
+@GENTOO_PORTAGE_EPREFIX@/usr/lib/python2.5/site-packages
 @end example
 
 If you indeed use this absolute path to install your shared library,
@@ -11658,7 +11668,7 @@
 prerequisites for the @file{Makefile} file itself even with @option{-n}
 (@pxref{Remaking Makefiles,,, make, The GNU Make Manual}).
 @item
-@code{make SHELL="/bin/bash -vx"} can help debug complex rules.
+@code{make SHELL="@GENTOO_PORTAGE_EPREFIX@/bin/bash -vx"} can help debug complex rules.
 @xref{The Make Macro SHELL,,, autoconf, The Autoconf Manual}, for some
 portability quirks associated with this construct.
 @end itemize
@@ -11690,7 +11700,7 @@
 The first version of the @command{automake} script looks as follows.
 
 @example
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 
 status=0
 
diff -urN automake-1.11.1.orig/lib/INSTALL automake-1.11.1/lib/INSTALL
--- automake-1.11.1.orig/lib/INSTALL	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/lib/INSTALL	2012-04-03 01:16:36.071259300 -0700
@@ -142,10 +142,11 @@
 ==================
 
    By default, `make install' installs the package's commands under
-`/usr/local/bin', include files under `/usr/local/include', etc.  You
-can specify an installation prefix other than `/usr/local' by giving
-`configure' the option `--prefix=PREFIX', where PREFIX must be an
-absolute file name.
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/bin', include files under
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/include', etc.  You can specify
+an installation prefix other than `@GENTOO_PORTAGE_EPREFIX@/usr/local'
+by giving `configure' the option `--prefix=PREFIX', where PREFIX must be
+an absolute file name.
 
    You can specify separate installation prefixes for
 architecture-specific files and architecture-independent files.  If you
@@ -306,7 +307,9 @@
 Unfortunately, this technique does not work for `CONFIG_SHELL' due to
 an Autoconf bug.  Until the bug is fixed you can use this workaround:
 
-     CONFIG_SHELL=/bin/bash /bin/bash ./configure CONFIG_SHELL=/bin/bash
+     CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash \
+     	@GENTOO_PORTAGE_EPREFIX@/bin/bash ./configure \
+	CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 `configure' Invocation
 ======================
diff -urN automake-1.11.1.orig/lib/config-ml.in automake-1.11.1/lib/config-ml.in
--- automake-1.11.1.orig/lib/config-ml.in	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/lib/config-ml.in	2012-04-03 01:10:04.376936100 -0700
@@ -100,7 +100,7 @@
 # configured for a particular host.
 
 Makefile=${ac_file-Makefile}
-ml_config_shell=${CONFIG_SHELL-/bin/sh}
+ml_config_shell=${CONFIG_SHELL-@GENTOO_PORTAGE_EPREFIX@/bin/sh}
 ml_realsrcdir=${srcdir}
 
 # Scan all the arguments and set all the ones we need.
diff -urN automake-1.11.1.orig/lib/config.guess automake-1.11.1/lib/config.guess
--- automake-1.11.1.orig/lib/config.guess	2012-04-03 00:03:58.529425200 -0700
+++ automake-1.11.1/lib/config.guess	2012-04-03 01:14:32.765103000 -0700
@@ -874,7 +874,7 @@
 	  EV67)  UNAME_MACHINE=alphaev67 ;;
 	  EV68*) UNAME_MACHINE=alphaev68 ;;
         esac
-	objdump --private-headers /bin/sh | grep -q ld.so.1
+	objdump --private-headers @GENTOO_PORTAGE_EPREFIX@/bin/sh | grep -q ld.so.1
 	if test "$?" = 0 ; then LIBC="libc1" ; else LIBC="" ; fi
 	echo ${UNAME_MACHINE}-unknown-linux-gnu${LIBC}
 	exit ;;
@@ -1027,7 +1027,7 @@
 	exit ;;
     i*86:*:5:[678]*)
     	# UnixWare 7.x, OpenUNIX and OpenServer 6.
-	case `/bin/uname -X | grep "^Machine"` in
+	case `@GENTOO_PORTAGE_EPREFIX@/bin/uname -X | grep "^Machine"` in
 	    *486*)	     UNAME_MACHINE=i486 ;;
 	    *Pentium)	     UNAME_MACHINE=i586 ;;
 	    *Pent*|*Celeron) UNAME_MACHINE=i686 ;;
@@ -1038,14 +1038,14 @@
 	if test -f /usr/options/cb.name; then
 		UNAME_REL=`sed -n 's/.*Version //p' </usr/options/cb.name`
 		echo ${UNAME_MACHINE}-pc-isc$UNAME_REL
-	elif /bin/uname -X 2>/dev/null >/dev/null ; then
-		UNAME_REL=`(/bin/uname -X|grep Release|sed -e 's/.*= //')`
-		(/bin/uname -X|grep i80486 >/dev/null) && UNAME_MACHINE=i486
-		(/bin/uname -X|grep '^Machine.*Pentium' >/dev/null) \
+	elif @GENTOO_PORTAGE_EPREFIX@/bin/uname -X 2>/dev/null >/dev/null ; then
+		UNAME_REL=`(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X|grep Release|sed -e 's/.*= //')`
+		(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X|grep i80486 >/dev/null) && UNAME_MACHINE=i486
+		(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X|grep '^Machine.*Pentium' >/dev/null) \
 			&& UNAME_MACHINE=i586
-		(/bin/uname -X|grep '^Machine.*Pent *II' >/dev/null) \
+		(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X|grep '^Machine.*Pent *II' >/dev/null) \
 			&& UNAME_MACHINE=i686
-		(/bin/uname -X|grep '^Machine.*Pentium Pro' >/dev/null) \
+		(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X|grep '^Machine.*Pentium Pro' >/dev/null) \
 			&& UNAME_MACHINE=i686
 		echo ${UNAME_MACHINE}-pc-sco$UNAME_REL
 	else
@@ -1090,12 +1090,12 @@
 	OS_REL=''
 	test -r /etc/.relid \
 	&& OS_REL=.`sed -n 's/[^ ]* [^ ]* \([0-9][0-9]\).*/\1/p' < /etc/.relid`
-	/bin/uname -p 2>/dev/null | grep 86 >/dev/null \
+	@GENTOO_PORTAGE_EPREFIX@/bin/uname -p 2>/dev/null | grep 86 >/dev/null \
 	  && { echo i486-ncr-sysv4.3${OS_REL}; exit; }
-	/bin/uname -p 2>/dev/null | /bin/grep entium >/dev/null \
+	@GENTOO_PORTAGE_EPREFIX@/bin/uname -p 2>/dev/null | @GENTOO_PORTAGE_EPREFIX@/bin/grep entium >/dev/null \
 	  && { echo i586-ncr-sysv4.3${OS_REL}; exit; } ;;
     3[34]??:*:4.0:* | 3[34]??,*:*:4.0:*)
-        /bin/uname -p 2>/dev/null | grep 86 >/dev/null \
+        @GENTOO_PORTAGE_EPREFIX@/bin/uname -p 2>/dev/null | grep 86 >/dev/null \
           && { echo i486-ncr-sysv4; exit; } ;;
     NCR*:*:4.2:* | MPRAS*:*:4.2:*)
 	OS_REL='.3'
@@ -1103,9 +1103,9 @@
 	    && OS_REL=.`sed -n 's/[^ ]* [^ ]* \([0-9][0-9]\).*/\1/p' < /etc/.relid`
 	/bin/uname -p 2>/dev/null | grep 86 >/dev/null \
 	    && { echo i486-ncr-sysv4.3${OS_REL}; exit; }
-	/bin/uname -p 2>/dev/null | /bin/grep entium >/dev/null \
+	/bin/uname -p 2>/dev/null | @GENTOO_PORTAGE_EPREFIX@/bin/grep entium >/dev/null \
 	    && { echo i586-ncr-sysv4.3${OS_REL}; exit; }
-	/bin/uname -p 2>/dev/null | /bin/grep pteron >/dev/null \
+	/bin/uname -p 2>/dev/null | @GENTOO_PORTAGE_EPREFIX@/bin/grep pteron >/dev/null \
 	    && { echo i586-ncr-sysv4.3${OS_REL}; exit; } ;;
     m68*:LynxOS:2.*:* | m68*:LynxOS:3.0*:*)
 	echo m68k-unknown-lynxos${UNAME_RELEASE}
@@ -1482,8 +1482,8 @@
 uname -s = `(uname -s) 2>/dev/null || echo unknown`
 uname -v = `(uname -v) 2>/dev/null || echo unknown`
 
-/usr/bin/uname -p = `(/usr/bin/uname -p) 2>/dev/null`
-/bin/uname -X     = `(/bin/uname -X) 2>/dev/null`
+@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p = `(@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p) 2>/dev/null`
+@GENTOO_PORTAGE_EPREFIX@/bin/uname -X     = `(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X) 2>/dev/null`
 
 hostinfo               = `(hostinfo) 2>/dev/null`
 /bin/universe          = `(/bin/universe) 2>/dev/null`
diff -urN automake-1.11.1.orig/lib/missing automake-1.11.1/lib/missing
--- automake-1.11.1.orig/lib/missing	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/lib/missing	2012-04-03 01:17:16.226788800 -0700
@@ -209,7 +209,7 @@
 	touch $file
     else
 	test -z "$file" || exec >$file
-	echo "#! /bin/sh"
+	echo "#! @GENTOO_PORTAGE_EPREFIX@/bin/sh"
 	echo "# Created by GNU Automake missing as a replacement of"
 	echo "#  $ $@"
 	echo "exit 0"
diff -urN automake-1.11.1.orig/m4/multi.m4 automake-1.11.1/m4/multi.m4
--- automake-1.11.1.orig/m4/multi.m4	2009-12-03 16:38:13.000000000 -0800
+++ automake-1.11.1/m4/multi.m4	2012-04-03 01:18:59.233886100 -0700
@@ -61,5 +61,5 @@
 with_target_subdir="$with_target_subdir"
 ac_configure_args="${multilib_arg} ${ac_configure_args}"
 multi_basedir="$multi_basedir"
-CONFIG_SHELL=${CONFIG_SHELL-/bin/sh}
+CONFIG_SHELL=${CONFIG_SHELL-@GENTOO_PORTAGE_EPREFIX@/bin/sh}
 CC="$CC"])])dnl
diff -urN automake-1.11.1.orig/tests/README automake-1.11.1/tests/README
--- automake-1.11.1.orig/tests/README	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/tests/README	2012-04-03 01:53:11.709799200 -0700
@@ -33,7 +33,7 @@
 Getting details from failures
 -----------------------------
 
-  Each test is a shell script, and by default is run by /bin/sh.
+  Each test is a shell script, and by default is run by @GENTOO_PORTAGE_EPREFIX@/bin/sh.
   In a non-VPATH build you can run them directly, they will be verbose.
   By default, verbose output of a test foo.test is retained in the log
   file foo.log.  A summary log is created in the file test-suite.log.
@@ -62,7 +62,7 @@
   Note that this problem does not occur if zsh is executed through a
   symlink with a basename of `sh', since in that case it starts
   in Bourne compatibility mode.  So you should be perfectly safe when
-  /bin/sh is zsh.
+  @GENTOO_PORTAGE_EPREFIX@/bin/sh is zsh.
 
 
 Reporting failures
diff -urN automake-1.11.1.orig/tests/ccnoco.test automake-1.11.1/tests/ccnoco.test
--- automake-1.11.1.orig/tests/ccnoco.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/ccnoco.test	2012-04-03 01:20:12.611558700 -0700
@@ -55,7 +55,7 @@
 END
 
 cat > Mycomp << 'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 
 case " "$* in
  *\ -c*\ -o* | *\ -o*\ -c*)
diff -urN automake-1.11.1.orig/tests/ccnoco3.test automake-1.11.1/tests/ccnoco3.test
--- automake-1.11.1.orig/tests/ccnoco3.test	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/tests/ccnoco3.test	2012-04-03 01:20:31.526340900 -0700
@@ -45,7 +45,7 @@
 END
 
 cat > Mycomp << 'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 
 case " "$* in
  *\ -c*\ -o* | *\ -o*\ -c*)
diff -urN automake-1.11.1.orig/tests/check10-p.test automake-1.11.1/tests/check10-p.test
--- automake-1.11.1.orig/tests/check10-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check10-p.test	2012-04-03 01:20:50.537142300 -0700
@@ -30,15 +30,15 @@
 END
 
 cat >>pass <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 cat >>fail <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 cat >>skip <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 chmod a+x pass fail skip
diff -urN automake-1.11.1.orig/tests/check10.test automake-1.11.1/tests/check10.test
--- automake-1.11.1.orig/tests/check10.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check10.test	2012-04-03 01:21:13.148663700 -0700
@@ -30,15 +30,15 @@
 END
 
 cat >>pass <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 cat >>fail <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 cat >>skip <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 chmod a+x pass fail skip
diff -urN automake-1.11.1.orig/tests/check11-p.test automake-1.11.1/tests/check11-p.test
--- automake-1.11.1.orig/tests/check11-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check11-p.test	2012-04-03 01:21:26.066746800 -0700
@@ -29,7 +29,7 @@
 END
 
 cat >>skip <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 chmod a+x skip
diff -urN automake-1.11.1.orig/tests/check11.test automake-1.11.1/tests/check11.test
--- automake-1.11.1.orig/tests/check11.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check11.test	2012-04-03 01:21:37.764085800 -0700
@@ -29,7 +29,7 @@
 END
 
 cat >>skip <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 chmod a+x skip
diff -urN automake-1.11.1.orig/tests/check2-p.test automake-1.11.1/tests/check2-p.test
--- automake-1.11.1.orig/tests/check2-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check2-p.test	2012-04-03 01:21:55.513635000 -0700
@@ -32,7 +32,7 @@
 TESTS = \
   subrun.sh
 subrun.sh:
-	(echo '#! /bin/sh'; echo 'dir/echo.sh') > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; echo 'dir/echo.sh') > $@
 	chmod +x $@
 CLEANFILES = subrun.sh
 END
@@ -40,7 +40,7 @@
 cat > dir/Makefile.am << 'END'
 check_SCRIPTS = echo.sh
 echo.sh:
-	(echo '#! /bin/sh'; echo 'echo Hello') > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; echo 'echo Hello') > $@
 	chmod +x $@
 CLEANFILES = echo.sh
 END
diff -urN automake-1.11.1.orig/tests/check2.test automake-1.11.1/tests/check2.test
--- automake-1.11.1.orig/tests/check2.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check2.test	2012-04-03 01:22:17.097951000 -0700
@@ -32,7 +32,7 @@
 TESTS = \
   subrun.sh
 subrun.sh:
-	(echo '#! /bin/sh'; echo 'dir/echo.sh') > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; echo 'dir/echo.sh') > $@
 	chmod +x $@
 CLEANFILES = subrun.sh
 END
@@ -40,7 +40,7 @@
 cat > dir/Makefile.am << 'END'
 check_SCRIPTS = echo.sh
 echo.sh:
-	(echo '#! /bin/sh'; echo 'echo Hello') > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; echo 'echo Hello') > $@
 	chmod +x $@
 CLEANFILES = echo.sh
 END
diff -urN automake-1.11.1.orig/tests/check3-p.test automake-1.11.1/tests/check3-p.test
--- automake-1.11.1.orig/tests/check3-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check3-p.test	2012-04-03 01:22:33.644759700 -0700
@@ -33,7 +33,7 @@
 SUBDIRS = dir
 TESTS = subrun.sh
 subrun.sh:
-	(echo '#! /bin/sh'; cat command1.inc) > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; cat command1.inc) > $@
 	chmod +x $@
 command1.inc:
 	echo 'dir/echo.sh' > $@
@@ -47,7 +47,7 @@
 ## The next line ensures that command1.inc has been built before
 ## recurring into the subdir.
 	test -f ../command1.inc
-	(echo '#! /bin/sh'; cat command2.inc) > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; cat command2.inc) > $@
 	chmod +x $@
 command2.inc:
 	echo 'echo Hello' > $@
diff -urN automake-1.11.1.orig/tests/check3.test automake-1.11.1/tests/check3.test
--- automake-1.11.1.orig/tests/check3.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check3.test	2012-04-03 01:22:42.350000400 -0700
@@ -33,7 +33,7 @@
 SUBDIRS = dir
 TESTS = subrun.sh
 subrun.sh:
-	(echo '#! /bin/sh'; cat command1.inc) > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; cat command1.inc) > $@
 	chmod +x $@
 command1.inc:
 	echo 'dir/echo.sh' > $@
@@ -47,7 +47,7 @@
 ## The next line ensures that command1.inc has been built before
 ## recurring into the subdir.
 	test -f ../command1.inc
-	(echo '#! /bin/sh'; cat command2.inc) > $@
+	(echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh'; cat command2.inc) > $@
 	chmod +x $@
 command2.inc:
 	echo 'echo Hello' > $@
diff -urN automake-1.11.1.orig/tests/check4-p.test automake-1.11.1/tests/check4-p.test
--- automake-1.11.1.orig/tests/check4-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check4-p.test	2012-04-03 01:23:01.093248300 -0700
@@ -35,12 +35,12 @@
 echo TESTS = fail.sh >dir/Makefile.am
 
 cat >ok.sh <<'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 :
 END
 
 cat >dir/fail.sh <<'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
diff -urN automake-1.11.1.orig/tests/check4.test automake-1.11.1/tests/check4.test
--- automake-1.11.1.orig/tests/check4.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check4.test	2012-04-03 01:23:09.302389800 -0700
@@ -35,12 +35,12 @@
 echo TESTS = fail.sh >dir/Makefile.am
 
 cat >ok.sh <<'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 :
 END
 
 cat >dir/fail.sh <<'END'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
diff -urN automake-1.11.1.orig/tests/check6-p.test automake-1.11.1/tests/check6-p.test
--- automake-1.11.1.orig/tests/check6-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check6-p.test	2012-04-03 01:23:23.144657700 -0700
@@ -32,7 +32,7 @@
 END
 
 cat >>a <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
diff -urN automake-1.11.1.orig/tests/check6.test automake-1.11.1/tests/check6.test
--- automake-1.11.1.orig/tests/check6.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check6.test	2012-04-03 01:23:28.361200800 -0700
@@ -32,7 +32,7 @@
 END
 
 cat >>a <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
diff -urN automake-1.11.1.orig/tests/check7-p.test automake-1.11.1/tests/check7-p.test
--- automake-1.11.1.orig/tests/check7-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check7-p.test	2012-04-03 01:23:43.210670100 -0700
@@ -36,7 +36,7 @@
 END
 
 cat >>b <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 chmod a+x b
diff -urN automake-1.11.1.orig/tests/check7.test automake-1.11.1/tests/check7.test
--- automake-1.11.1.orig/tests/check7.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check7.test	2012-04-03 01:23:38.217671700 -0700
@@ -36,7 +36,7 @@
 END
 
 cat >>b <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 chmod a+x b
diff -urN automake-1.11.1.orig/tests/check8-p.test automake-1.11.1/tests/check8-p.test
--- automake-1.11.1.orig/tests/check8-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check8-p.test	2012-04-03 01:24:05.855698200 -0700
@@ -36,11 +36,11 @@
 mkdir sub
 
 cat >>foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 test -f "$srcdir/Makefile.am"
 END
 cat >>sub/foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 test -f "$srcdir/Makefile.am"
 END
 chmod a+x foo sub/foo
diff -urN automake-1.11.1.orig/tests/check8.test automake-1.11.1/tests/check8.test
--- automake-1.11.1.orig/tests/check8.test	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/tests/check8.test	2012-04-03 01:23:59.804488200 -0700
@@ -36,11 +36,11 @@
 mkdir sub
 
 cat >>foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 test -f "$srcdir/Makefile.am"
 END
 cat >>sub/foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 test -f "$srcdir/Makefile.am"
 END
 chmod a+x foo sub/foo
diff -urN automake-1.11.1.orig/tests/check9-p.test automake-1.11.1/tests/check9-p.test
--- automake-1.11.1.orig/tests/check9-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/check9-p.test	2012-04-03 01:41:35.008986900 -0700
@@ -38,11 +38,11 @@
 END
 
 cat >>pass-script.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 cat >>xfail-script.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 chmod a+x pass-script.test xfail-script.test
diff -urN automake-1.11.1.orig/tests/check9.test automake-1.11.1/tests/check9.test
--- automake-1.11.1.orig/tests/check9.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/check9.test	2012-04-03 01:41:13.037093400 -0700
@@ -38,11 +38,11 @@
 END
 
 cat >>pass-script.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 cat >>xfail-script.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 chmod a+x pass-script.test xfail-script.test
diff -urN automake-1.11.1.orig/tests/color-p.test automake-1.11.1/tests/color-p.test
--- automake-1.11.1.orig/tests/color-p.test	2009-12-08 10:17:17.000000000 -0800
+++ automake-1.11.1/tests/color-p.test	2012-04-03 01:42:14.264836500 -0700
@@ -49,17 +49,17 @@
 END
 
 cat >pass <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 
 cat >fail <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
 cat >skip <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 
@@ -94,7 +94,7 @@
 }
 
 cat >expect-make <<'END'
-#! /usr/bin/expect -f
+#! @GENTOO_PORTAGE_EPREFIX@/usr/bin/expect -f
 spawn $env(MAKE) -e check
 expect eof
 END
diff -urN automake-1.11.1.orig/tests/color.test automake-1.11.1/tests/color.test
--- automake-1.11.1.orig/tests/color.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/color.test	2012-04-03 01:42:29.371357200 -0700
@@ -49,17 +49,17 @@
 END
 
 cat >pass <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 
 cat >fail <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 END
 
 cat >skip <<END
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 77
 END
 
@@ -94,7 +94,7 @@
 }
 
 cat >expect-make <<'END'
-#! /usr/bin/expect -f
+#! @GENTOO_PORTAGE_EPREFIX@/usr/bin/expect -f
 spawn $env(MAKE) -e check
 expect eof
 END
diff -urN automake-1.11.1.orig/tests/compile2.test automake-1.11.1/tests/compile2.test
--- automake-1.11.1.orig/tests/compile2.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/compile2.test	2012-04-03 01:43:04.228827300 -0700
@@ -67,8 +67,8 @@
 # Do not actually run this test on anything that could be w32.
 test -d "C:\\" && Exit 77
 # This test is taken from Autoconf's _AS_PATH_SEPARATOR_PREPARE.
-(PATH='/bin;/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
-  (PATH='/bin:/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
+(PATH='@GENTOO_PORTAGE_EPREFIX@/bin;@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
+  (PATH='@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
     Exit 77
 }
 
diff -urN automake-1.11.1.orig/tests/cond14.test automake-1.11.1/tests/cond14.test
--- automake-1.11.1.orig/tests/cond14.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/cond14.test	2012-04-03 01:43:25.867154100 -0700
@@ -35,7 +35,7 @@
 bin_SCRIPTS = helldl
 helldl$(EXEEXT):
 	rm -f $@
-	echo '#! /bin/sh' > $@
+	echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh' > $@
 	echo '-dlopen is unsupported' >> $@
 	chmod +x $@
 endif
diff -urN automake-1.11.1.orig/tests/cond15.test automake-1.11.1/tests/cond15.test
--- automake-1.11.1.orig/tests/cond15.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/cond15.test	2012-04-03 01:43:39.467873700 -0700
@@ -31,7 +31,7 @@
 bin_SCRIPTS = helldl
 helldl$(EXEEXT):
 	rm -f $@
-	echo '#! /bin/sh' > $@
+	echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh' > $@
 	echo '-dlopen is unsupported' >> $@
 	chmod +x $@
 endif
@@ -41,7 +41,7 @@
 bin_SCRIPTS = helldl
 helldl$(EXEEXT):
 	rm -f $@
-	echo '#! /bin/sh' > $@
+	echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/sh' > $@
 	echo '-dlopen is unsupported' >> $@
 	chmod +x $@
 endif
diff -urN automake-1.11.1.orig/tests/cond39.test automake-1.11.1/tests/cond39.test
--- automake-1.11.1.orig/tests/cond39.test	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/tests/cond39.test	2012-04-03 01:44:09.202319400 -0700
@@ -69,7 +69,7 @@
 END
 
 cat >prog1.in <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 bindir='@bindir@'
 echo "hi, this is $0, and bindir is $bindir"
 END
diff -urN automake-1.11.1.orig/tests/dejagnu3.test automake-1.11.1/tests/dejagnu3.test
--- automake-1.11.1.orig/tests/dejagnu3.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/dejagnu3.test	2012-04-03 01:44:38.166111000 -0700
@@ -22,7 +22,7 @@
 set -e
 
 cat > hammer << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Everything looks like a nail to me!"
 END
 
diff -urN automake-1.11.1.orig/tests/dejagnu4.test automake-1.11.1/tests/dejagnu4.test
--- automake-1.11.1.orig/tests/dejagnu4.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/dejagnu4.test	2012-04-03 01:44:53.961769500 -0700
@@ -23,14 +23,14 @@
 set -e
 
 cat > hammer << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Everything looks like a nail to me!"
 END
 
 chmod +x hammer
 
 cat > spanner << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "I'm a right spanner!"
 END
 
diff -urN automake-1.11.1.orig/tests/dejagnu5.test automake-1.11.1/tests/dejagnu5.test
--- automake-1.11.1.orig/tests/dejagnu5.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/dejagnu5.test	2012-04-03 01:45:16.814839200 -0700
@@ -26,7 +26,7 @@
 package=`$FGREP AC_INIT configure.in | sed -e 's:^[^[]*\[\([^]]*\)\].*$:\1:'`
 
 cat > $package << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Ah, we have been expecting you, Mr. Blond."
 END
 
diff -urN automake-1.11.1.orig/tests/dejagnu6.test automake-1.11.1/tests/dejagnu6.test
--- automake-1.11.1.orig/tests/dejagnu6.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/dejagnu6.test	2012-04-03 01:45:27.746525100 -0700
@@ -22,7 +22,7 @@
 set -e
 
 cat > faildeja << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo whatever
 END
 
diff -urN automake-1.11.1.orig/tests/dejagnu7.test automake-1.11.1/tests/dejagnu7.test
--- automake-1.11.1.orig/tests/dejagnu7.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/dejagnu7.test	2012-04-03 01:45:33.747225000 -0700
@@ -24,7 +24,7 @@
 runtest --help | $FGREP -e --status || Exit 77
 
 cat > failtcl << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo whatever
 END
 
diff -urN automake-1.11.1.orig/tests/gnits2.test automake-1.11.1/tests/gnits2.test
--- automake-1.11.1.orig/tests/gnits2.test	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/tests/gnits2.test	2012-04-03 01:46:33.166106400 -0700
@@ -63,14 +63,14 @@
 mkdir sub
 
 cat >sub/scriptok.sh <<EOF
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Which version? Which usage?"
 EOF
 
 # Not only does this script not support --help/--version, but
 # it will also hang when run without input.
 cat >sub/scriptnok.sh <<EOF
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 cat
 EOF
 
diff -urN automake-1.11.1.orig/tests/gnits3.test automake-1.11.1/tests/gnits3.test
--- automake-1.11.1.orig/tests/gnits3.test	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/tests/gnits3.test	2012-04-03 01:46:42.385950000 -0700
@@ -51,7 +51,7 @@
 mkdir sub
 
 cat >nok.sh <<EOF
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Which version? Which usage?"
 exit 1
 EOF
diff -urN automake-1.11.1.orig/tests/instmany-mans.test automake-1.11.1/tests/instmany-mans.test
--- automake-1.11.1.orig/tests/instmany-mans.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/instmany-mans.test	2012-04-03 01:47:09.155802900 -0700
@@ -40,7 +40,7 @@
   done; }`
 
 sed "s|@limit@|$limit|g" >myinstall.in <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Fake install script.  This doesn't really install
 # (the INSTALL path below would be wrong outside this directory).
 limit=@limit@
@@ -55,7 +55,7 @@
 
 # Creative quoting in the next line to please maintainer-check.
 sed "s|@limit@|$limit|g" >'rm' <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 limit=@limit@
 PATH=$save_PATH
 export PATH
diff -urN automake-1.11.1.orig/tests/instmany-python.test automake-1.11.1/tests/instmany-python.test
--- automake-1.11.1.orig/tests/instmany-python.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/instmany-python.test	2012-04-03 01:47:35.533077300 -0700
@@ -35,7 +35,7 @@
   done; }`
 
 sed "s|@limit@|$limit|g" >myinstall.in <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Fake install script.  This doesn't really install
 # (the INSTALL path below would be wrong outside this directory).
 limit=@limit@
@@ -50,7 +50,7 @@
 
 # Creative quoting in the next line to please maintainer-check.
 sed "s|@limit@|$limit|g" >'rm' <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 limit=@limit@
 PATH=$save_PATH
 export PATH
diff -urN automake-1.11.1.orig/tests/instmany.test automake-1.11.1/tests/instmany.test
--- automake-1.11.1.orig/tests/instmany.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/instmany.test	2012-04-03 01:47:55.229015700 -0700
@@ -48,7 +48,7 @@
   done; }`
 
 sed "s|@limit@|$limit|g" >myinstall.in <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Fake install script.  This doesn't really install
 # (the INSTALL path below would be wrong outside this directory).
 limit=@limit@
@@ -63,7 +63,7 @@
 
 # Creative quoting in the next line to please maintainer-check.
 sed "s|@limit@|$limit|g" >'rm' <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 limit=@limit@
 PATH=$save_PATH
 export PATH
diff -urN automake-1.11.1.orig/tests/man4.test automake-1.11.1/tests/man4.test
--- automake-1.11.1.orig/tests/man4.test	2009-12-08 10:04:46.000000000 -0800
+++ automake-1.11.1/tests/man4.test	2012-04-03 01:48:39.685405200 -0700
@@ -53,7 +53,7 @@
 END
 
 cat > foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 while test $# -gt 0; do
   case $1 in
     -h | --help) echo "usage: $0 [OPTIONS]..."; exit 0;;
@@ -66,7 +66,7 @@
 cp foo bar
 
 cat > help2man <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # fake help2man script that lets `missing' think it is not installed
 exit 127
 END
diff -urN automake-1.11.1.orig/tests/missing5.test automake-1.11.1/tests/missing5.test
--- automake-1.11.1.orig/tests/missing5.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/missing5.test	2012-04-03 01:48:57.770521500 -0700
@@ -36,7 +36,7 @@
 
 for tool in $needed_tools; do
   cat >$tool.in <<EOF
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exec @$tool@ "\$@"
 EOF
 done
diff -urN automake-1.11.1.orig/tests/mkinst3.test automake-1.11.1/tests/mkinst3.test
--- automake-1.11.1.orig/tests/mkinst3.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/mkinst3.test	2012-04-03 01:49:15.341535000 -0700
@@ -37,7 +37,7 @@
 
 mkdir bin
 cat >bin/mkdir <<'EOF'
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 for arg
 do
   case $arg in
diff -urN automake-1.11.1.orig/tests/mmodely.test automake-1.11.1/tests/mmodely.test
--- automake-1.11.1.orig/tests/mmodely.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/mmodely.test	2012-04-03 01:49:48.325630500 -0700
@@ -75,7 +75,7 @@
 $MAKE
 
 cat >myyacc.sh <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "$@" >y.tab.c
 END
 cat >mylex.sh <<'END'
diff -urN automake-1.11.1.orig/tests/multlib.test automake-1.11.1/tests/multlib.test
--- automake-1.11.1.orig/tests/multlib.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/multlib.test	2012-04-03 01:50:02.486462100 -0700
@@ -34,7 +34,7 @@
 END
 
 cat >mycc <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 case ${1+"$@"} in
  *-print-multi-lib*)
   echo ".;"
diff -urN automake-1.11.1.orig/tests/parallel-tests.test automake-1.11.1/tests/parallel-tests.test
--- automake-1.11.1.orig/tests/parallel-tests.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests.test	2012-04-03 01:50:29.852934300 -0700
@@ -41,17 +41,17 @@
 END
 
 cat >>foo.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 0
 END
 cat >>bar.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 99
 END
 cat >>baz.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 1
 END
diff -urN automake-1.11.1.orig/tests/parallel-tests10.test automake-1.11.1/tests/parallel-tests10.test
--- automake-1.11.1.orig/tests/parallel-tests10.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests10.test	2012-04-03 01:50:48.064575900 -0700
@@ -31,7 +31,7 @@
 END
 
 cat >foo <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 
diff -urN automake-1.11.1.orig/tests/parallel-tests2.test automake-1.11.1/tests/parallel-tests2.test
--- automake-1.11.1.orig/tests/parallel-tests2.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests2.test	2012-04-03 01:51:03.956753700 -0700
@@ -37,18 +37,18 @@
 END
 
 cat >>foo.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 test -f bla || exit 1
 exit 0
 END
 cat >>bar.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 99
 END
 cat >>baz.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 1
 END
diff -urN automake-1.11.1.orig/tests/parallel-tests3.test automake-1.11.1/tests/parallel-tests3.test
--- automake-1.11.1.orig/tests/parallel-tests3.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests3.test	2012-04-03 01:51:19.448851500 -0700
@@ -38,7 +38,7 @@
 for i in 1 2 3 4 5 6 7 8; do
   echo "TESTS += foo$i.test" >> Makefile.am
   cat >foo$i.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 # hack around maintainer-check check:
 sleep='sleep '1
diff -urN automake-1.11.1.orig/tests/parallel-tests4.test automake-1.11.1/tests/parallel-tests4.test
--- automake-1.11.1.orig/tests/parallel-tests4.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests4.test	2012-04-03 01:51:26.519765400 -0700
@@ -36,7 +36,7 @@
 mkdir sub
 
 cat >foo.chk << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 chmod a+x foo.chk
diff -urN automake-1.11.1.orig/tests/parallel-tests5.test automake-1.11.1/tests/parallel-tests5.test
--- automake-1.11.1.orig/tests/parallel-tests5.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests5.test	2012-04-03 01:51:35.289519000 -0700
@@ -43,7 +43,7 @@
   echo "scripts += foo$i.test" >> Makefile.am
   echo "programs += foo$i.prog" >> Makefile.am
   cat >foo$i.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 0
 END
diff -urN automake-1.11.1.orig/tests/parallel-tests7.test automake-1.11.1/tests/parallel-tests7.test
--- automake-1.11.1.orig/tests/parallel-tests7.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests7.test	2012-04-03 01:51:55.293018900 -0700
@@ -45,7 +45,7 @@
 mkdir sub
 
 cat >chk-driver <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo $0 "$@"
 shift
 shift
@@ -57,7 +57,7 @@
 cp chk-driver noext-driver
 
 cat >foo.chk << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 END
 chmod a+x foo.chk
diff -urN automake-1.11.1.orig/tests/parallel-tests8.test automake-1.11.1/tests/parallel-tests8.test
--- automake-1.11.1.orig/tests/parallel-tests8.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests8.test	2012-04-03 01:52:03.313122600 -0700
@@ -45,7 +45,7 @@
 END
 
 cat >>foo.in <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 0
 END
diff -urN automake-1.11.1.orig/tests/parallel-tests9.test automake-1.11.1/tests/parallel-tests9.test
--- automake-1.11.1.orig/tests/parallel-tests9.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/parallel-tests9.test	2012-04-03 01:52:11.233706400 -0700
@@ -35,18 +35,18 @@
 END
 
 cat >>foo.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 test -f bla || exit 1
 exit 0
 END
 cat >>bar.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 99
 END
 cat >>baz.test <<'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "this is $0"
 exit 1
 END
diff -urN automake-1.11.1.orig/tests/pr287.test automake-1.11.1/tests/pr287.test
--- automake-1.11.1.orig/tests/pr287.test	2009-12-08 10:02:32.000000000 -0800
+++ automake-1.11.1/tests/pr287.test	2012-04-03 01:52:34.292817300 -0700
@@ -32,7 +32,7 @@
 END
 
 cat > foo << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo Maude
 END
 
diff -urN automake-1.11.1.orig/tests/strip.test automake-1.11.1/tests/strip.test
--- automake-1.11.1.orig/tests/strip.test	2009-12-08 10:02:33.000000000 -0800
+++ automake-1.11.1/tests/strip.test	2012-04-03 01:53:33.584673300 -0700
@@ -32,7 +32,7 @@
 END
 
 cat > foo << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo Maude
 END
 
diff -urN automake-1.11.1.orig/tests/vala4.test automake-1.11.1/tests/vala4.test
--- automake-1.11.1.orig/tests/vala4.test	2009-12-08 10:15:40.000000000 -0800
+++ automake-1.11.1/tests/vala4.test	2012-04-03 01:54:19.257806100 -0700
@@ -35,7 +35,7 @@
 : > Makefile.am
 
 cat > valac << 'END'
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 if test "x$1" = x--version; then
   echo 1.2.3
 fi
