--- a/pym/portage/dbapi/vartree.py	2012-03-08 01:07:01.234904000 -0800
+++ b/pym/portage/dbapi/vartree.py	2012-03-08 01:08:16.801404000 -0800
@@ -252,10 +252,10 @@
 			if self._fs_lock_obj is not None:
 				raise AssertionError("already locked")
 			try:
-				self._fs_lock_obj = lockfile(self._conf_mem_file)
+				self._fs_lock_obj = lockfile(self._conf_mem_file, 1)
 			except InvalidLocation:
 				self.settings._init_dirs()
-				self._fs_lock_obj = lockfile(self._conf_mem_file)
+				self._fs_lock_obj = lockfile(self._conf_mem_file, 1)
 		self._fs_lock_count += 1
 
 	def _fs_unlock(self):
