diff -urN work.orig/config.guess work/config.guess
--- work.orig/config.guess	2012-04-06 13:09:09.600294200 -0700
+++ work/config.guess	2012-04-06 13:38:13.225383800 -0700
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Attempt to guess a canonical system name.
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
@@ -908,7 +908,7 @@
 	  EV67)  UNAME_MACHINE=alphaev67 ;;
 	  EV68*) UNAME_MACHINE=alphaev68 ;;
 	esac
-	objdump --private-headers /bin/sh | grep -q ld.so.1
+	objdump --private-headers @GENTOO_PORTAGE_EPREFIX@/bin/sh | grep -q ld.so.1
 	if test "$?" = 0 ; then LIBC="gnulibc1" ; fi
 	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
 	exit ;;
@@ -1529,6 +1529,8 @@
 /usr/bin/uname -p = `(/usr/bin/uname -p) 2>/dev/null`
 /bin/uname -X     = `(/bin/uname -X) 2>/dev/null`
 
+@GENTOO_PORTAGE_EPREFIX@/bin/uname -a = `(@GENTOO_PORTAGE_EPREFIX@/bin/uname -a) 2>/dev/null`
+
 hostinfo               = `(hostinfo) 2>/dev/null`
 /bin/universe          = `(/bin/universe) 2>/dev/null`
 /usr/bin/arch -k       = `(/usr/bin/arch -k) 2>/dev/null`
diff -urN work.orig/config.sub work/config.sub
--- work.orig/config.sub	2012-04-06 13:09:09.182252400 -0700
+++ work/config.sub	2012-04-06 13:41:06.427624300 -0700
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Configuration validation subroutine script.
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
diff -urN work.orig/testsuite/config-guess.sh work/testsuite/config-guess.sh
--- work.orig/testsuite/config-guess.sh	2012-03-03 14:19:41.000000000 -0800
+++ work/testsuite/config-guess.sh	2012-04-06 13:42:56.899875200 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 #
 # Copyright 2004, 2005, 2009 Free Software Foundation, Inc.
 # Contributed by Ben Elliston <bje@gnu.org>.
diff -urN work.orig/testsuite/config-sub.sh work/testsuite/config-sub.sh
--- work.orig/testsuite/config-sub.sh	2012-03-03 14:19:41.000000000 -0800
+++ work/testsuite/config-sub.sh	2012-04-06 13:42:21.062974100 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 #
 # Copyright 2004, 2005, 2009 Free Software Foundation, Inc.
 # Contributed by Ben Elliston <bje@gnu.org>.
diff -urN work.orig/testsuite/uname.in work/testsuite/uname.in
--- work.orig/testsuite/uname.in	2012-03-03 14:19:41.000000000 -0800
+++ work/testsuite/uname.in	2012-04-06 13:42:06.781275600 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 # uname(1) simulator, inspired by Pavel Roskin.
 
 [ $# -ne 1 ] && exec sh $0 -s
diff -urN work.orig/uname work/uname
--- work.orig/uname	2012-03-03 14:19:41.000000000 -0800
+++ work/uname	2012-04-06 13:41:35.195229600 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 # uname(1) simulator, inspired by Pavel Roskin.
 
 [ $# -ne 1 ] && exec sh $0 -s
