diff -urN readline-6.2.orig/examples/Makefile.in readline-6.2/examples/Makefile.in
--- readline-6.2.orig/examples/Makefile.in	2010-12-28 12:55:32.000000000 -0800
+++ readline-6.2/examples/Makefile.in	2012-08-17 01:17:06.075484000 -0700
@@ -40,6 +40,7 @@
 INSTALL_DATA = @INSTALL_DATA@
 
 EXEEXT = @EXEEXT@
+EXE = $(EXEEXT)
 OBJEXT = @OBJEXT@
 
 # Support an alternate destination root directory for package building
@@ -54,13 +55,24 @@
 INCLUDES = -I$(srcdir) -I$(top_srcdir) -I..
 
 CCFLAGS  = $(DEFS) $(LOCAL_CFLAGS) $(CPPFLAGS) $(INCLUDES) $(CFLAGS)
-LDFLAGS = -g -L.. @LDFLAGS@
+LDFLAGS = -g -L../shlib -L.. @LDFLAGS@
 
 PURIFY = @PURIFY@
 
-READLINE_LIB = ../libreadline.a
-HISTORY_LIB = ../libhistory.a
-
+SHLIB_DOT = @SHLIB_DOT@
+SHLIB_LIBPREF = @SHLIB_LIBPREF@
+SHLIB_LIBSTUFF = @SHLIB_LIBSTUFF@
+
+SHLIB_LIBVERSION = @SHLIB_LIBVERSION@
+SHLIB_DLLVERSION = @SHLIB_DLLVERSION@
+
+SHLIB_MAJOR=		@SHLIB_MAJOR@
+SHLIB_MINOR=		.@SHLIB_MINOR@
+
+READLINE_LIB = ../shlib/$(SHLIB_LIBPREF)readline$(SHLIB_DOT)$(SHLIB_LIBVERSION)
+HISTORY_LIB = ../shlib/$(SHLIB_LIBPREF)history$(SHLIB_DOT)$(SHLIB_LIBVERSION)
+READLINE_LIB_STAT = ../libreadline.a
+HISTORY_LIB_STAT = ../libhistory.a
 TERMCAP_LIB = @TERMCAP_LIB@
 
 .c.o:
