diff -urN libtool-2.4.orig/configure.ac libtool-2.4/configure.ac
--- libtool-2.4.orig/configure.ac	2010-09-22 07:43:26.000000000 -0700
+++ libtool-2.4/configure.ac	2012-04-04 13:21:55.491581500 -0700
@@ -128,6 +128,7 @@
 dnl much older releases of autoconf and automake.  See release notes.
 dnl 1.11 is needed for color-tests, 1.11.1 fixes a security issue.
 AM_INIT_AUTOMAKE([1.11.1 gnu subdir-objects dist-xz color-tests parallel-tests])
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------------- ##
diff -urN libtool-2.4.orig/libltdl/configure.ac libtool-2.4/libltdl/configure.ac
--- libtool-2.4.orig/libltdl/configure.ac	2010-09-22 07:43:26.000000000 -0700
+++ libtool-2.4/libltdl/configure.ac	2012-04-04 13:22:26.426267200 -0700
@@ -53,6 +53,7 @@
 ## ------------------------ ##
 
 AM_INIT_AUTOMAKE([gnu])
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------------------- ##
diff -urN libtool-2.4.orig/tests/cdemo/configure.ac libtool-2.4/tests/cdemo/configure.ac
--- libtool-2.4.orig/tests/cdemo/configure.ac	2010-08-23 08:31:21.000000000 -0700
+++ libtool-2.4/tests/cdemo/configure.ac	2012-04-04 13:31:53.201099500 -0700
@@ -35,6 +35,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/demo/configure.ac libtool-2.4/tests/demo/configure.ac
--- libtool-2.4.orig/tests/demo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/demo/configure.ac	2012-04-04 13:32:11.071672900 -0700
@@ -36,6 +36,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ---------------- ##
diff -urN libtool-2.4.orig/tests/depdemo/configure.ac libtool-2.4/tests/depdemo/configure.ac
--- libtool-2.4.orig/tests/depdemo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/depdemo/configure.ac	2012-04-04 13:32:27.860029900 -0700
@@ -36,6 +36,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ---------------- ##
diff -urN libtool-2.4.orig/tests/f77demo/configure.ac libtool-2.4/tests/f77demo/configure.ac
--- libtool-2.4.orig/tests/f77demo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/f77demo/configure.ac	2012-04-04 13:32:48.998256700 -0700
@@ -38,6 +38,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/fcdemo/configure.ac libtool-2.4/tests/fcdemo/configure.ac
--- libtool-2.4.orig/tests/fcdemo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/fcdemo/configure.ac	2012-04-04 13:33:06.170690500 -0700
@@ -38,6 +38,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/mdemo/configure.ac libtool-2.4/tests/mdemo/configure.ac
--- libtool-2.4.orig/tests/mdemo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/mdemo/configure.ac	2012-04-04 13:33:26.157687100 -0700
@@ -36,6 +36,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/mdemo2/configure.ac libtool-2.4/tests/mdemo2/configure.ac
--- libtool-2.4.orig/tests/mdemo2/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/mdemo2/configure.ac	2012-04-04 13:33:39.216798400 -0700
@@ -36,6 +36,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/pdemo/configure.ac libtool-2.4/tests/pdemo/configure.ac
--- libtool-2.4.orig/tests/pdemo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/pdemo/configure.ac	2012-04-04 13:33:56.804815300 -0700
@@ -36,6 +36,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ------------------ ##
diff -urN libtool-2.4.orig/tests/tagdemo/configure.ac libtool-2.4/tests/tagdemo/configure.ac
--- libtool-2.4.orig/tests/tagdemo/configure.ac	2010-08-23 08:31:22.000000000 -0700
+++ libtool-2.4/tests/tagdemo/configure.ac	2012-04-04 13:34:14.536360900 -0700
@@ -38,6 +38,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE([disable])
 
 
 ## ---------------- ##
