diff -urN libtool-2.4.2.orig/libltdl/config/general.m4sh libtool-2.4.2/libltdl/config/general.m4sh
--- libtool-2.4.2.orig/libltdl/config/general.m4sh	2012-04-08 16:13:16.886320600 -0700
+++ libtool-2.4.2/libltdl/config/general.m4sh	2012-04-08 16:15:06.893320600 -0700
@@ -557,8 +557,12 @@
     func_quote_for_expand_result="$my_arg"
 }
 
+func_update_status ()
+{
+    return $1
+}
 
-# func_show_eval cmd [fail_exp]
+# func_show_eval cmd [fail_exp] [quote-for-eval-yesno (dfl: no)]
 # Unless opt_silent is true, then output CMD.  Then, if opt_dryrun is
 # not true, evaluate CMD.  If the evaluation of CMD fails, and FAIL_EXP
 # is given, then evaluate it.
@@ -566,23 +570,34 @@
 {
     my_cmd="$1"
     my_fail_exp="${2-:}"
+    my_show_vars="${3-no}"
+    if test x$my_show_vars = x ; then
+	    my_show_vars=no
+    fi
 
     ${opt_silent-false} || {
-      func_quote_for_expand "$my_cmd"
-      eval "func_echo $func_quote_for_expand_result"
+      if test $my_show_vars = yes ; then
+	func_quote_for_eval "$my_cmd"
+	eval "func_echo $func_quote_for_eval"
+	eval "func_echo $func_quote_for_eval_result"
+      else
+	func_quote_for_expand "$my_cmd"
+        eval "func_echo $func_quote_for_expand_result"
+        eval "func_echo $func_quote_for_expand_result"
+      fi
     }
 
     if ${opt_dry_run-false}; then :; else
       eval "$my_cmd"
       my_status=$?
       if test "$my_status" -eq 0; then :; else
-	eval "(exit $my_status); $my_fail_exp"
+	eval "func_update_status $my_status; $my_fail_exp"
       fi
     fi
 }
 
 
-# func_show_eval_locale cmd [fail_exp]
+# func_show_eval_locale cmd [fail_exp] [quote-for-eval-yesno (dfl: no)]
 # Unless opt_silent is true, then output CMD.  Then, if opt_dryrun is
 # not true, evaluate CMD.  If the evaluation of CMD fails, and FAIL_EXP
 # is given, then evaluate it.  Use the saved locale for evaluation.
@@ -590,10 +605,18 @@
 {
     my_cmd="$1"
     my_fail_exp="${2-:}"
-
+    my_show_vars="${3-no}"
+    if test x$my_show_vars = x ; then
+	    my_show_vars=no
+    fi
     ${opt_silent-false} || {
-      func_quote_for_expand "$my_cmd"
-      eval "func_echo $func_quote_for_expand_result"
+      if test $my_show_vars = yes ; then
+	func_quote_for_eval "$my_cmd"
+	eval "func_echo $func_quote_for_eval_result"
+      else
+        func_quote_for_expand "$my_cmd"
+        eval "func_echo $func_quote_for_expand_result"
+      fi
     }
 
     if ${opt_dry_run-false}; then :; else
@@ -602,7 +625,7 @@
       my_status=$?
       eval "$lt_safe_locale"
       if test "$my_status" -eq 0; then :; else
-	eval "(exit $my_status); $my_fail_exp"
+	eval "func_update_status $my_status; $my_fail_exp"
       fi
     fi
 }
diff -urN libtool-2.4.2.orig/libltdl/config/ltmain.m4sh libtool-2.4.2/libltdl/config/ltmain.m4sh
--- libtool-2.4.2.orig/libltdl/config/ltmain.m4sh	2012-04-08 16:13:16.896820600 -0700
+++ libtool-2.4.2/libltdl/config/ltmain.m4sh	2012-04-08 16:14:49.631320600 -0700
@@ -789,6 +789,7 @@
 {
   $opt_debug
   # awkward: cmd appends spaces to result
+  # FIXME: does this subshell serve an indispensible purpose?
   func_convert_core_msys_to_w32_result=`( cmd //c echo "$1" ) 2>/dev/null |
     $SED -e 's/[ ]*$//' -e "$lt_sed_naive_backslashify"`
 }
@@ -2224,8 +2225,11 @@
 	    # so we also need to try rm && ln -s.
 	    for linkname
 	    do
-	      test "$linkname" != "$realname" \
-		&& func_show_eval "(cd $destdir && { $LN_S -f $realname $linkname || { $RM $linkname && $LN_S $realname $linkname; }; })"
+	      if test "$linkname" != "$realname" ; then
+		func_relative_path "`pwd`" "$destdir"
+		linkname_eval="\"$func_relative_path_result$linkname\""
+		func_show_eval "$LN_S -f $realname $linkname_eval || { $RM $linkname_eval && $LN_S $realname $linkname_eval ; }"
+	      fi
 	    done
 	  fi
 
@@ -2707,7 +2711,7 @@
 	done
 
 	# Now compile the dynamic symbol file.
-	func_show_eval '(cd $output_objdir && $LTCC$symtab_cflags -c$no_builtin_flag$pic_flag_for_symtable "$my_dlsyms")' 'exit $?'
+	func_show_eval "set_result_1 () { return \$1 ; } ; cd \"$output_objdir\" && { $LTCC$symtab_cflags -c$no_builtin_flag$pic_flag_for_symtable $my_dlsyms ; cached_result=\$? ; cd \"`pwd`\" ; set_result_1 \$cached_result ; }" 'exit $?' yes
 
 	# Clean up the generated files.
 	func_show_eval '$RM "$output_objdir/$my_dlsyms" "$nlist" "${nlist}S" "${nlist}T"'
@@ -2935,12 +2939,12 @@
 	sleep 2
       done
     fi
-    func_show_eval "(cd \$f_ex_an_ar_dir && $AR x \"\$f_ex_an_ar_oldlib\")" \
-		   'stat=$?; rm -f "$lockfile"; exit $stat'
+    func_show_eval "set_result_2 () { return \$1 ; } ; cd \"$f_ex_an_ar_dir\" && { $AR x \"$f_ex_an_ar_oldlib\" ; cached_result=\$? ; cd \"`pwd`\" ; set_result_2 \$cached_result ; }" \
+		   'stat=$?; rm -f "$lockfile"; exit $stat' yes
     if test "$lock_old_archive_extraction" = yes; then
       $opt_dry_run || rm -f "$lockfile"
     fi
-    if ($AR t "$f_ex_an_ar_oldlib" | sort | sort -uc >/dev/null 2>&1); then
+    if { $AR t "$f_ex_an_ar_oldlib" | sort | sort -uc >/dev/null 2>&1 ; } ; then
      :
     else
       func_fatal_error "object name conflicts in archive: $f_ex_an_ar_dir/$f_ex_an_ar_oldlib"
@@ -3082,7 +3086,7 @@
 
 # The HP-UX ksh and POSIX shell print the target directory to stdout
 # if CDPATH is set.
-(unset CDPATH) >/dev/null 2>&1 && unset CDPATH
+unset CDPATH >/dev/null 2>&1 || :
 
 relink_command=\"$relink_command\"
 
@@ -7525,7 +7529,9 @@
 
 	# Make a backup of the uninstalled library when relinking
 	if test "$opt_mode" = relink; then
-	  $opt_dry_run || eval '(cd $output_objdir && $RM ${realname}U && $MV $realname ${realname}U)' || exit $?
+	  func_relative_path "`pwd`" "$output_objdir"
+	  realname_eval="\"$func_relative_path_result$realname\""
+	  $opt_dry_run || eval "$RM ${realname_eval}U && $MV $realname_eval ${realname_eval}U" || exit $? 
 	fi
 
 	# Do each of the archive commands.
@@ -7697,9 +7703,9 @@
 
 		# Restore the uninstalled library and exit
 		if test "$opt_mode" = relink; then
-		  ( cd "$output_objdir" && \
-		    $RM "${realname}T" && \
-		    $MV "${realname}U" "$realname" )
+		  cd "$output_objdir" && \
+		  $RM "${realname}T" && \
+		  $MV "${realname}U" "$realname"
 		fi
 
 		exit $lt_exit
@@ -7790,9 +7796,9 @@
 
 	    # Restore the uninstalled library and exit
 	    if test "$opt_mode" = relink; then
-	      ( cd "$output_objdir" && \
-	        $RM "${realname}T" && \
-		$MV "${realname}U" "$realname" )
+	      cd "$output_objdir" && \
+	      $RM "${realname}T" && \
+	      $MV "${realname}U" "$realname"
 	    fi
 
 	    exit $lt_exit
@@ -7802,8 +7808,11 @@
 
 	# Restore the uninstalled library and exit
 	if test "$opt_mode" = relink; then
-	  $opt_dry_run || eval '(cd $output_objdir && $RM ${realname}T && $MV $realname ${realname}T && $MV ${realname}U $realname)' || exit $?
-
+	  $opt_dry_run || { 
+	    func_relative_path "`pwd`" "$output_objdir"
+	    realname_eval="\"$func_relative_path_result$realname\""
+	    eval "$RM ${realname_eval}T && $MV $realname_eval ${realname_eval}T && $MV ${realname_eval}U $realname_eval" || exit $0
+	  }
 	  if test -n "$convenience"; then
 	    if test -z "$whole_archive_flag_spec"; then
 	      func_show_eval '${RM}r "$gentop"'
@@ -7816,7 +7825,10 @@
 	# Create links to the real library.
 	for linkname in $linknames; do
 	  if test "$realname" != "$linkname"; then
-	    func_show_eval '(cd "$output_objdir" && $RM "$linkname" && $LN_S "$realname" "$linkname")' 'exit $?'
+
+	    func_relative_path "`pwd`" "$output_objdir"
+	    link_targ_eval="\"$func_relative_path_result$linkname\""
+	    func_show_eval "$RM $link_targ_eval && $LN_S \"../$realname\" $link_targ_eval" 'exit $?'
 	  fi
 	done
 
@@ -8262,7 +8274,7 @@
 	    relink_command="$var=$func_quote_for_eval_result; export $var; $relink_command"
 	  fi
 	done
-	relink_command="(cd `pwd`; $relink_command)"
+	relink_command="set_result_6 () { return \$1 ; } ; oldpwd_6=\`pwd\` ; cd \"`pwd`\" ; $relink_command ; cached_result=\$? ; cd \"\$oldpwd_6\" ; set_result_6 \$cached_result"
 	relink_command=`$ECHO "$relink_command" | $SED "$sed_quote_subst"`
       fi
 
@@ -8516,7 +8528,7 @@
 	fi
       done
       # Quote the link command for shipping.
-      relink_command="(cd `pwd`; $SHELL $progpath $preserve_args --mode=relink $libtool_args @inst_prefix_dir@)"
+      relink_command="set_result_7 () { return \$1 ; } ; oldpwd_7=\`pwd\` ; cd \"`pwd`\" ; $SHELL $progpath $preserve_args --mode=relink $libtool_args @inst_prefix_dir@ ; cached_result=\$? ; cd \"\$oldpwd_7\" ; set_result_7 \$cached_result"
       relink_command=`$ECHO "$relink_command" | $SED "$sed_quote_subst"`
       if test "$hardcode_automatic" = yes ; then
 	relink_command=
@@ -8686,7 +8698,9 @@
 
       # Do a symbolic link so that the libtool archive can be found in
       # LD_LIBRARY_PATH before the program is installed.
-      func_show_eval '( cd "$output_objdir" && $RM "$outputname" && $LN_S "../$outputname" "$outputname" )' 'exit $?'
+      func_relative_path "`pwd`" "$output_objdir"
+      link_targ_eval="\"$func_relative_path_result$outputname\""
+      func_show_eval "$RM $link_targ_eval && $LN_S \"../$outputname\" $link_targ_eval" 'exit $?'
       ;;
     esac
     exit $EXIT_SUCCESS
