From 1f3c3f46e36c4d90ba6fbccc9b44f2e2814759d4 Mon Sep 17 00:00:00 2001
From: Hib Eris <hib@hiberis.nl>
Date: Thu, 03 Nov 2011 11:48:51 +0000
Subject: Fix build for platforms with executable extensions

---
diff --git a/Makefile.am b/Makefile.am
index bbde408..16dcf8b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -48,7 +48,7 @@ $(pkgconfig_DATA): config.status
 
 install-data-hook: install-binPROGRAMS
 if ENABLE_UPDATE_MIMEDB
-	$(DESTDIR)"$(bindir)/update-mime-database" -V "$(DESTDIR)$(datadir)/mime"
+	$(DESTDIR)"$(bindir)/update-mime-database$(EXEEXT)" -V "$(DESTDIR)$(datadir)/mime"
 endif
 
 uninstall-hook:
@@ -78,21 +78,21 @@ all: $(defaultmakedeps)
 create-pot:
 	$(AM_V_GEN) $(MAKE) -C po shared-mime-info.pot
 
-check: create-pot freedesktop.org.xml update-mime-database check-translations
+check: create-pot freedesktop.org.xml update-mime-database$(EXEEXT) check-translations
 	if test -e $(top_builddir)/freedesktop.org.xml; then \
 		xmllint --noout --valid $(top_builddir)/freedesktop.org.xml || exit 1;	\
 	else \
 		xmllint --noout --valid $(top_srcdir)/freedesktop.org.xml || exit 1;	\
 	fi; \
-	if test -d .git/ && test -x $(top_builddir)/update-mime-database && test -d $(top_srcdir)/mime-db-tests ; then \
-		$(top_builddir)/update-mime-database mime-db-tests/ ; \
+	if test -d .git/ && test -x $(top_builddir)/update-mime-database$(EXEEXT) && test -d $(top_srcdir)/mime-db-tests ; then \
+		$(top_builddir)/update-mime-database$(EXEEXT) mime-db-tests/ ; \
 	fi
-	if test -d .git/ && test -x ../xdgmime/src/test-mime-data && test -x $(top_builddir)/test-tree-magic ; then \
+	if test -d .git/ && test -x ../xdgmime/src/test-mime-data$(EXEEXT) && test -x $(top_builddir)/test-tree-magic$(EXEEXT) ; then \
 		mkdir -p $(top_builddir)/temp-mime-dir/mime/packages ; \
 		cp -a $(top_builddir)/freedesktop.org.xml $(top_builddir)/temp-mime-dir/mime/packages/ ; \
-		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" $(top_builddir)/update-mime-database "$(top_builddir)/temp-mime-dir/mime/" ; \
-		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" XDG_DATA_HOME="$(top_builddir)/temp-mime-dir/" ../xdgmime/src/test-mime-data -v tests/list || (XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" ../xdgmime/src/test-mime-data -v -v tests/list ; exit 1); \
-		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" $(top_builddir)/test-tree-magic tests/tree-list || exit 1; \
+		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" $(top_builddir)/update-mime-database$(EXEEXT) "$(top_builddir)/temp-mime-dir/mime/" ; \
+		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" XDG_DATA_HOME="$(top_builddir)/temp-mime-dir/" ../xdgmime/src/test-mime-data$(EXEEXT) -v tests/list || (XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" ../xdgmime/src/test-mime-data$(EXEEXT) -v -v tests/list ; exit 1); \
+		XDG_DATA_DIRS="$(top_builddir)/temp-mime-dir/" $(top_builddir)/test-tree-magic$(EXEEXT) tests/tree-list || exit 1; \
 		rm -rf "$(top_builddir)/temp-mime-dir/" ; \
 	fi
 	@if `grep comment $(srcdir)/freedesktop.org.xml.in | grep -v _comment | grep -q -v '<!'` ; then \
--
cgit v0.9.0.2-2-gbebe
