diff -urN autoconf-2.68.orig/GNUmakefile autoconf-2.68/GNUmakefile
--- autoconf-2.68.orig/GNUmakefile	2010-09-21 14:58:04.000000000 -0700
+++ autoconf-2.68/GNUmakefile	2012-04-03 04:47:20.126723700 -0700
@@ -20,10 +20,10 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-# Systems where /bin/sh is not the default shell need this.  The $(shell)
+# Systems where @GENTOO_PORTAGE_EPREFIX@/bin/sh is not the default shell need this.  The $(shell)
 # command below won't work with e.g. stock DOS/Windows shells.
-ifeq ($(wildcard /bin/s[h]),/bin/sh)
-SHELL = /bin/sh
+ifeq ($(wildcard @GENTOO_PORTAGE_EPREFIX@/bin/s[h]),@GENTOO_PORTAGE_EPREFIX@/bin/sh)
+SHELL = @GENTOO_PORTAGE_EPREFIX@/bin/sh
 else
 # will be used only with the next shell-test line, then overwritten
 # by a configured-in value
diff -urN autoconf-2.68.orig/INSTALL autoconf-2.68/INSTALL
--- autoconf-2.68.orig/INSTALL	2010-09-22 15:35:57.000000000 -0700
+++ autoconf-2.68/INSTALL	2012-04-03 04:48:27.577723700 -0700
@@ -142,7 +142,8 @@
 ==================
 
    By default, `make install' installs the package's commands under
-`/usr/local/bin', include files under `/usr/local/include', etc.  You
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/bin', include files under
+`@GENTOO_PORTAGE_EPREFIX@/usr/local/include', etc.  You
 can specify an installation prefix other than `/usr/local' by giving
 `configure' the option `--prefix=PREFIX', where PREFIX must be an
 absolute file name.
@@ -306,7 +307,8 @@
 Unfortunately, this technique does not work for `CONFIG_SHELL' due to
 an Autoconf bug.  Until the bug is fixed you can use this workaround:
 
-     CONFIG_SHELL=/bin/bash /bin/bash ./configure CONFIG_SHELL=/bin/bash
+     CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash @GENTOO_PORTAGE_EPREFIX@/bin/bash \
+     	./configure CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 `configure' Invocation
 ======================
diff -urN autoconf-2.68.orig/configure.ac autoconf-2.68/configure.ac
--- autoconf-2.68.orig/configure.ac	2010-09-20 12:51:44.000000000 -0700
+++ autoconf-2.68/configure.ac	2012-04-03 04:18:19.818223700 -0700
@@ -53,13 +53,13 @@
 # Also, some implementations of /bin/sh (e.g., Solaris 8) are soooo slow
 # that they are unusable on large scripts like our testsuite.
 
-# So before using `/bin/sh -n' to check our scripts, we first check
-# that `/bin/sh -n' is known to not have these problems.
+# So before using `@GENTOO_PORTAGE_EPREFIX@/bin/sh -n' to check our scripts, we first check
+# that `@GENTOO_PORTAGE_EPREFIX@/bin/sh -n' is known to not have these problems.
 
-AC_CACHE_CHECK([whether /bin/sh -n is known to work], [ac_cv_sh_n_works],
+AC_CACHE_CHECK([whether @GENTOO_PORTAGE_EPREFIX@/bin/sh -n is known to work], [ac_cv_sh_n_works],
 [if (
     unset BASH_VERSION ZSH_VERSION
-    /bin/sh -c '
+    @GENTOO_PORTAGE_EPREFIX@/bin/sh -c '
       test -n "${BASH_VERSION+set}" || # Bash
       test -n "${KSH_VERSION+set}" || # pdksh
       test -n "${ZSH_VERSION+set}" || # zsh
diff -urN autoconf-2.68.orig/doc/autoconf.texi autoconf-2.68/doc/autoconf.texi
--- autoconf-2.68.orig/doc/autoconf.texi	2010-09-21 13:14:56.000000000 -0700
+++ autoconf-2.68/doc/autoconf.texi	2012-04-03 04:43:58.847723700 -0700
@@ -862,7 +862,7 @@
 the standard targets, the substitutions provided by Autoconf, automatic
 dependency tracking, @code{VPATH} building, and so on.  @command{make}
 builds the @code{hello} program, and @code{make install} installs it
-in @file{/usr/local/bin} (or whatever prefix was given to
+in @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin} (or whatever prefix was given to
 @command{configure}, if not @file{/usr/local}).
 
 The benefits of Automake increase for larger packages (especially ones
@@ -2002,7 +2002,7 @@
 produces this in @command{configure}:
 
 @example
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 # From configure.ac Revision: 1.30
 @end example
 @end defmac
@@ -2487,7 +2487,7 @@
 example, an input shell script should begin like this:
 
 @example
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 # @@configure_input@@
 @end example
 
@@ -8267,7 +8267,7 @@
 @defmac AC_SYS_INTERPRETER
 @acindex{SYS_INTERPRETER}
 Check whether the system supports starting scripts with a line of the
-form @samp{#!/bin/sh} to select the interpreter to use for the script.
+form @samp{#!@GENTOO_PORTAGE_EPREFIX@/bin/sh} to select the interpreter to use for the script.
 After running this macro, shell code in @file{configure.ac} can check
 the shell variable @code{interpval}; it is set to @samp{yes}
 if the system supports @samp{#!}, @samp{no} if not.
@@ -9634,7 +9634,7 @@
 @example
 $ @kbd{CC=/usr/bin/cc ./configure var=raboof --silent}
 $ @kbd{./config.status --recheck}
-running CONFIG_SHELL=/bin/sh /bin/sh ./configure var=raboof \
+running CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/sh @GENTOO_PORTAGE_EPREFIX@/bin/sh ./configure var=raboof \
   CC=/usr/bin/cc  --no-create --no-recursion
 @end example
 @end itemize
@@ -13678,7 +13678,7 @@
 @evindex LC_ALL
 @evindex SHELL
 Initialize the M4sh environment.  This macro calls @code{m4_init}, then
-outputs the @code{#! /bin/sh} line, a notice about where the output was
+outputs the @code{#! @GENTOO_PORTAGE_EPREFIX@/bin/sh} line, a notice about where the output was
 generated from, and code to sanitize the environment for the rest of the
 script.  Among other initializations, this sets @env{SHELL} to the shell
 chosen to run the script (@pxref{CONFIG_SHELL}), and @env{LC_ALL} to
@@ -19369,7 +19369,7 @@
 Not all @command{make} implementations define this @code{SHELL} macro.
 Tru64
 @command{make} is an example; this implementation always uses
-@code{/bin/sh}.  So it's a good idea to always define @code{SHELL} in
+@code{@GENTOO_PORTAGE_EPREFIX@/bin/sh}.  So it's a good idea to always define @code{SHELL} in
 your makefiles.  If you use Autoconf, do
 
 @example
@@ -21868,9 +21868,9 @@
 @evindex CONFIG_SHELL
 The shell with which to run @command{configure} for the @option{--recheck}
 option.  It must be Bourne-compatible.  The default is a shell that
-supports @code{LINENO} if available, and @file{/bin/sh} otherwise.
+supports @code{LINENO} if available, and @file{@GENTOO_PORTAGE_EPREFIX@/bin/sh} otherwise.
 Invoking @command{configure} by hand bypasses this setting, so you may
-need to use a command like @samp{CONFIG_SHELL=/bin/bash /bin/bash ./configure}
+need to use a command like @samp{CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash @GENTOO_PORTAGE_EPREFIX@/bin/bash ./configure}
 to insure that the same shell is used everywhere.  The absolute name of the
 shell should be passed.
 @end defvar
diff -urN autoconf-2.68.orig/doc/install.texi autoconf-2.68/doc/install.texi
--- autoconf-2.68.orig/doc/install.texi	2010-06-18 08:04:28.000000000 -0700
+++ autoconf-2.68/doc/install.texi	2012-04-03 04:45:44.778723700 -0700
@@ -168,9 +168,9 @@
 @section Installation Names
 
 By default, @samp{make install} installs the package's commands under
-@file{/usr/local/bin}, include files under @file{/usr/local/include}, etc.
+@file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin}, include files under @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/include}, etc.
 You can specify an
-installation prefix other than @file{/usr/local} by giving
+installation prefix other than @file{@GENTOO_PORTAGE_EPREFIX@/usr/local} by giving
 @command{configure} the option @option{--prefix=@var{prefix}}, where
 @var{prefix} must be an absolute file name.
 
@@ -364,7 +364,7 @@
 workaround:
 
 @example
-CONFIG_SHELL=/bin/bash /bin/bash ./configure CONFIG_SHELL=/bin/bash
+CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash @GENTOO_PORTAGE_EPREFIX@/bin/bash ./configure CONFIG_SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/bash
 @end example
 
 @node configure Invocation
diff -urN autoconf-2.68.orig/doc/make-stds.texi autoconf-2.68/doc/make-stds.texi
--- autoconf-2.68.orig/doc/make-stds.texi	2010-09-21 14:58:04.000000000 -0700
+++ autoconf-2.68/doc/make-stds.texi	2012-04-03 04:46:26.713723700 -0700
@@ -54,7 +54,7 @@
 Every Makefile should contain this line:
 
 @example
-SHELL = /bin/sh
+SHELL = @GENTOO_PORTAGE_EPREFIX@/bin/sh
 @end example
 
 @noindent
@@ -412,7 +412,7 @@
 @item bindir
 @vindex bindir
 The directory for installing executable programs that users can run.
-This should normally be @file{/usr/local/bin}, but write it as
+This should normally be @file{@GENTOO_PORTAGE_EPREFIX@/usr/local/bin}, but write it as
 @file{$(exec_prefix)/bin}.
 (If you are using Autoconf, write it as @samp{@@bindir@@}.)
 
diff -urN autoconf-2.68.orig/lib/autoconf/erlang.m4 autoconf-2.68/lib/autoconf/erlang.m4
--- autoconf-2.68.orig/lib/autoconf/erlang.m4	2010-09-16 14:10:08.000000000 -0700
+++ autoconf-2.68/lib/autoconf/erlang.m4	2012-04-03 04:48:50.314223700 -0700
@@ -102,7 +102,7 @@
 AC_LANG_DEFINE([Erlang], [erl], [ERL], [ERLC], [],
 [ac_ext=erl
 ac_compile='$ERLC $ERLCFLAGS -b beam conftest.$ac_ext >&AS_MESSAGE_LOG_FD'
-ac_link='$ERLC $ERLCFLAGS -b beam conftest.$ac_ext >&AS_MESSAGE_LOG_FD && echo "[#]!/bin/sh" > conftest$ac_exeext && AS_ECHO(["\"$ERL\" -run conftest start -run init stop -noshell"]) >> conftest$ac_exeext && chmod +x conftest$ac_exeext'
+ac_link='$ERLC $ERLCFLAGS -b beam conftest.$ac_ext >&AS_MESSAGE_LOG_FD && echo "[#]!@GENTOO_PORTAGE_EPREFIX@/bin/sh" > conftest$ac_exeext && AS_ECHO(["\"$ERL\" -run conftest start -run init stop -noshell"]) >> conftest$ac_exeext && chmod +x conftest$ac_exeext'
 ])
 
 
diff -urN autoconf-2.68.orig/lib/autoconf/general.m4 autoconf-2.68/lib/autoconf/general.m4
--- autoconf-2.68.orig/lib/autoconf/general.m4	2010-09-22 14:38:27.000000000 -0700
+++ autoconf-2.68/lib/autoconf/general.m4	2012-04-03 04:50:09.387223700 -0700
@@ -46,7 +46,7 @@
 # it reaches the end of `configure.ac'.
 #
 # - BINSH
-#   #! /bin/sh
+#   #! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 # - HEADER-REVISION
 #   Sent by AC_REVISION
 # - HEADER-COMMENT
@@ -589,7 +589,7 @@
 AC_SUBST([sharedstatedir], ['${prefix}/com'])dnl
 AC_SUBST([localstatedir],  ['${prefix}/var'])dnl
 AC_SUBST([includedir],     ['${prefix}/include'])dnl
-AC_SUBST([oldincludedir],  ['/usr/include'])dnl
+AC_SUBST([oldincludedir],  ['@GENTOO_PORTAGE_EPREFIX@/usr/include'])dnl
 AC_SUBST([docdir],         [m4_ifset([AC_PACKAGE_TARNAME],
 				     ['${datarootdir}/doc/${PACKAGE_TARNAME}'],
 				     ['${datarootdir}/doc/${PACKAGE}'])])dnl
diff -urN autoconf-2.68.orig/lib/autoconf/programs.m4 autoconf-2.68/lib/autoconf/programs.m4
--- autoconf-2.68.orig/lib/autoconf/programs.m4	2010-09-16 14:10:08.000000000 -0700
+++ autoconf-2.68/lib/autoconf/programs.m4	2012-04-03 04:51:31.565723700 -0700
@@ -547,7 +547,9 @@
 [[# Account for people who put trailing slashes in PATH elements.
 case $as_dir/ in @%:@((
   ./ | .// | /[cC]/* | \
-  /etc/* | /usr/sbin/* | /usr/etc/* | /sbin/* | /usr/afsws/bin/* | \
+  @GENTOO_PORTAGE_EPREFIX@/etc/* | @GENTOO_PORTAGE_EPREFIX@/usr/sbin/* | \
+  @GENTOO_PORTAGE_EPREFIX@/usr/etc/* | @GENTOO_PORTAGE_EPREFIX@/sbin/* | \
+  /usr/afsws/bin/* | \
   ?:[\\/]os2[\\/]install[\\/]* | ?:[\\/]OS2[\\/]INSTALL[\\/]* | \
   /usr/ucb/* ) ;;
   *)]
@@ -822,7 +824,7 @@
 ac_make=`AS_ECHO(["$[2]"]) | sed 's/+/p/g; s/[[^a-zA-Z0-9_]]/_/g'`
 AC_CACHE_VAL(ac_cv_prog_make_${ac_make}_set,
 [cat >conftest.make <<\_ACEOF
-SHELL = /bin/sh
+SHELL = @GENTOO_PORTAGE_EPREFIX@/bin/sh
 all:
 	@echo '@@@%%%=$(MAKE)=@@@%%%'
 _ACEOF
diff -urN autoconf-2.68.orig/lib/autoconf/specific.m4 autoconf-2.68/lib/autoconf/specific.m4
--- autoconf-2.68.orig/lib/autoconf/specific.m4	2010-09-16 14:10:08.000000000 -0700
+++ autoconf-2.68/lib/autoconf/specific.m4	2012-04-03 04:51:54.299223700 -0700
@@ -60,11 +60,11 @@
 # ------------------
 AC_DEFUN([AC_SYS_INTERPRETER],
 [AC_CACHE_CHECK(whether @%:@! works in shell scripts, ac_cv_sys_interpreter,
-[echo '#! /bin/cat
+[echo '#! @GENTOO_PORTAGE_EPREFIX@/bin/cat
 exit 69
 ' >conftest
 chmod u+x conftest
-(SHELL=/bin/sh; export SHELL; ./conftest >/dev/null 2>&1)
+(SHELL=@GENTOO_PORTAGE_EPREFIX@/bin/sh; export SHELL; ./conftest >/dev/null 2>&1)
 if test $? -ne 69; then
    ac_cv_sys_interpreter=yes
 else
diff -urN autoconf-2.68.orig/lib/autotest/general.m4 autoconf-2.68/lib/autotest/general.m4
--- autoconf-2.68.orig/lib/autotest/general.m4	2010-09-16 14:10:08.000000000 -0700
+++ autoconf-2.68/lib/autotest/general.m4	2012-04-03 04:52:23.704223700 -0700
@@ -218,7 +218,7 @@
 unlimited permission to copy, distribute and modify it.], [m4_echo])]
 [AS_PREPARE
 
-SHELL=${CONFIG_SHELL-/bin/sh}
+SHELL=${CONFIG_SHELL-@GENTOO_PORTAGE_EPREFIX@/bin/sh}
 
 # How were we run?
 at_cli_args="$[@]"
@@ -363,7 +363,7 @@
 at_fn_create_debugging_script ()
 {
   {
-    echo "#! /bin/sh" &&
+    echo "#! @GENTOO_PORTAGE_EPREFIX@/bin/sh" &&
     echo 'test "${ZSH_VERSION+set}" = set dnl
 && alias -g '\''${1+"$[@]"}'\''='\''"$[@]"'\''' &&
     AS_ECHO(["cd '$at_dir'"]) &&
diff -urN autoconf-2.68.orig/lib/m4sugar/m4sh.m4 autoconf-2.68/lib/m4sugar/m4sh.m4
--- autoconf-2.68.orig/lib/m4sugar/m4sh.m4	2010-09-16 15:44:55.000000000 -0700
+++ autoconf-2.68/lib/m4sugar/m4sh.m4	2012-04-03 04:54:06.319723700 -0700
@@ -37,7 +37,7 @@
 # KILL is only used to suppress output.
 #
 # - BINSH
-#   AC_REQUIRE'd #! /bin/sh line
+#   AC_REQUIRE'd #! @GENTOO_PORTAGE_EPREFIX@/bin/sh line
 # - HEADER-REVISION
 #   RCS keywords etc.
 # - HEADER-COMMENT
@@ -205,7 +205,7 @@
 	[as_have_required=no])
   AS_IF([test x$as_have_required = xyes && _AS_RUN(["$as_suggested"])],
     [],
-    [_AS_PATH_WALK([/bin$PATH_SEPARATOR/usr/bin$PATH_SEPARATOR$PATH],
+    [_AS_PATH_WALK([@GENTOO_PORTAGE_EPREFIX@/bin$PATH_SEPARATOR@GENTOO_PORTAGE_EPREFIX@/usr/bin$PATH_SEPARATOR$PATH],
       [case $as_dir in @%:@(
 	 /*)
 	   for as_base in sh bash ksh sh5; do
@@ -257,7 +257,7 @@
   fi
   AS_EXIT])])
 fi
-SHELL=${CONFIG_SHELL-/bin/sh}
+SHELL=${CONFIG_SHELL-@GENTOO_PORTAGE_EPREFIX@/bin/sh}
 export SHELL
 # Unset more variables known to interfere with behavior of common tools.
 CLICOLOR_FORCE= GREP_OPTIONS=
@@ -1277,8 +1277,8 @@
 [# The user is always right.
 if test "${PATH_SEPARATOR+set}" != set; then
   PATH_SEPARATOR=:
-  (PATH='/bin;/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
-    (PATH='/bin:/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
+  (PATH='@GENTOO_PORTAGE_EPREFIX@/bin;@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 && {
+    (PATH='@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/bin'; FPATH=$PATH; sh -c :) >/dev/null 2>&1 ||
       PATH_SEPARATOR=';'
   }
 fi
@@ -1656,8 +1656,8 @@
 uname -s = `(uname -s) 2>/dev/null || echo unknown`
 uname -v = `(uname -v) 2>/dev/null || echo unknown`
 
-/usr/bin/uname -p = `(/usr/bin/uname -p) 2>/dev/null || echo unknown`
-/bin/uname -X     = `(/bin/uname -X) 2>/dev/null     || echo unknown`
+@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p = `(@GENTOO_PORTAGE_EPREFIX@/usr/bin/uname -p) 2>/dev/null || echo unknown`
+@GENTOO_PORTAGE_EPREFIX@/bin/uname -X     = `(@GENTOO_PORTAGE_EPREFIX@/bin/uname -X) 2>/dev/null     || echo unknown`
 
 /bin/arch              = `(/bin/arch) 2>/dev/null              || echo unknown`
 /usr/bin/arch -k       = `(/usr/bin/arch -k) 2>/dev/null       || echo unknown`
@@ -2135,7 +2135,7 @@
 m4_pattern_forbid([^_?AS_])
 
 # Bangshe and minimal initialization.
-m4_divert_text([BINSH], [@%:@! /bin/sh])
+m4_divert_text([BINSH], [@%:@! @GENTOO_PORTAGE_EPREFIX@/bin/sh])
 m4_divert_text([HEADER-COMMENT],
 	       [@%:@ Generated from __file__ by m4_PACKAGE_STRING.])
 m4_divert_text([M4SH-SANITIZE], [_AS_SHELL_SANITIZE])
diff -urN autoconf-2.68.orig/tests/autotest.at autoconf-2.68/tests/autotest.at
--- autoconf-2.68.orig/tests/autotest.at	2010-09-13 12:28:28.000000000 -0700
+++ autoconf-2.68/tests/autotest.at	2012-04-03 04:55:08.510223700 -0700
@@ -1638,7 +1638,7 @@
 [AT_SKIP_PARALLEL_TESTS
 mkdir bin
 cat >bin/mkfifo <<\EOF
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 1
 EOF
 chmod +x bin/mkfifo
diff -urN autoconf-2.68.orig/tests/c.at autoconf-2.68/tests/c.at
--- autoconf-2.68.orig/tests/c.at	2010-08-25 16:19:42.000000000 -0700
+++ autoconf-2.68/tests/c.at	2012-04-03 04:55:36.589223700 -0700
@@ -109,7 +109,7 @@
 AT_SETUP([AC_PROG_CPP with warnings])
 
 AT_DATA([mycpp],
-[[#! /bin/sh
+[[#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo noise >&2
 exec "$@"
 ]])
@@ -146,7 +146,7 @@
 
 # A cpp which exit status is meaningless.
 AT_DATA([mycpp],
-[[#! /bin/sh
+[[#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 /lib/cpp "$@"
 exit 0
 ]])
@@ -185,7 +185,7 @@
   [], [ignore], [ignore])
 
 AT_DATA([mycc],
-[[#! /bin/sh
+[[#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 echo "Annoying copyright message" >&2
 exec "$@"
 ]])
diff -urN autoconf-2.68.orig/tests/local.at autoconf-2.68/tests/local.at
--- autoconf-2.68.orig/tests/local.at	2010-08-14 11:00:57.000000000 -0700
+++ autoconf-2.68/tests/local.at	2012-04-03 04:56:17.292223700 -0700
@@ -51,7 +51,7 @@
 # otherwise, do nothing.
 m4_define([AT_CHECK_SHELL_SYNTAX],
 [AT_SKIP_IF([test "$ac_cv_sh_n_works" != yes])
-AT_CHECK([/bin/sh -n $1])])
+AT_CHECK([@GENTOO_PORTAGE_EPREFIX@/bin/sh -n $1])])
 
 m4_define([AT_CHECK_PERL_SYNTAX],
 [AT_CHECK([autom4te_perllibdir=$abs_top_srcdir/lib $PERL -c "$abs_top_builddir"/bin/$1],
diff -urN autoconf-2.68.orig/tests/semantics.at autoconf-2.68/tests/semantics.at
--- autoconf-2.68.orig/tests/semantics.at	2010-09-22 14:55:00.000000000 -0700
+++ autoconf-2.68/tests/semantics.at	2012-04-03 04:57:12.027723700 -0700
@@ -484,7 +484,7 @@
 [mkdir path
 
 cat >path/tool <<\EOF
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 exit 0
 EOF
 chmod +x path/tool
diff -urN autoconf-2.68.orig/tests/tools.at autoconf-2.68/tests/tools.at
--- autoconf-2.68.orig/tests/tools.at	2010-09-13 12:28:38.000000000 -0700
+++ autoconf-2.68/tests/tools.at	2012-04-03 04:57:40.270223700 -0700
@@ -23,8 +23,8 @@
 ## Syntax of the shell scripts.  ##
 ## ----------------------------- ##
 
-# We use `/bin/sh -n script' to check that there are no syntax errors
-# in the scripts.  Although incredible, there are /bin/sh that go into
+# We use `@GENTOO_PORTAGE_EPREFIX@/bin/sh -n script' to check that there are no syntax errors
+# in the scripts.  Although incredible, there are @GENTOO_PORTAGE_EPREFIX@/bin/sh that go into
 # endless loops with `-n', e.g., SunOS's:
 #
 #   $ uname -a
@@ -40,8 +40,8 @@
 #   $ time sh -nx endless.sh
 #   ^Csh -nx endless.sh  3,67s user 0,03s system 63% cpu 5,868 total
 #
-# So before using `/bin/sh -n' to check our scripts, we first check
-# that `/bin/sh -n' is not broken to death.
+# So before using `@GENTOO_PORTAGE_EPREFIX@/bin/sh -n' to check our scripts, we first check
+# that `@GENTOO_PORTAGE_EPREFIX@/bin/sh -n' is not broken to death.
 
 AT_SETUP([Syntax of the shell scripts])
 
diff -urN autoconf-2.68.orig/tests/torture.at autoconf-2.68/tests/torture.at
--- autoconf-2.68.orig/tests/torture.at	2010-09-21 13:16:38.000000000 -0700
+++ autoconf-2.68/tests/torture.at	2012-04-03 04:58:42.715723700 -0700
@@ -1629,11 +1629,11 @@
 	 0, [ignore], [stderr],
 	 [AT_CHECK([grep 'rerun with a POSIX shell' stderr], [], [ignore])])
 chmod a-w .
-AT_CHECK([{ /bin/sh ./configure $configure_options --help=recursive; chmod u+w .; } | grep " INNERMOST "],
+AT_CHECK([{ @GENTOO_PORTAGE_EPREFIX@/bin/sh ./configure $configure_options --help=recursive; chmod u+w .; } | grep " INNERMOST "],
 	 0, [ignore], [stderr],
 	 [AT_CHECK([grep 'rerun with a POSIX shell' stderr], [], [ignore])])
 chmod a-w .
-AT_CHECK([PATH=.$PATH_SEPARATOR$PATH; export PATH; { /bin/sh configure $configure_options --help=recursive; chmod +w .; } | grep " INNERMOST "],
+AT_CHECK([PATH=.$PATH_SEPARATOR$PATH; export PATH; { @GENTOO_PORTAGE_EPREFIX@/bin/sh configure $configure_options --help=recursive; chmod +w .; } | grep " INNERMOST "],
 	 0, [ignore], [stderr],
 	 [AT_CHECK([grep 'rerun with a POSIX shell' stderr], [], [ignore])])
 chmod a-w .
@@ -1671,13 +1671,13 @@
 	 0, [ignore], [stderr],
 	 [AT_CHECK([grep 'rerun with a POSIX shell' stderr], [], [ignore])])
 chmod a-w builddir
-AT_CHECK([cd builddir && { /bin/sh ../configure $configure_options --help=recursive; chmod u+w .; } | grep " INNERMOST "],
+AT_CHECK([cd builddir && { @GENTOO_PORTAGE_EPREFIX@/bin/sh ../configure $configure_options --help=recursive; chmod u+w .; } | grep " INNERMOST "],
 	 0, [ignore], [stderr],
 	 [AT_CHECK([grep 'rerun with a POSIX shell' stderr], [], [ignore])])
 chmod a-w builddir
 # Not all shells search $PATH for scripts.
-if (cd builddir && PATH=`pwd`/..$PATH_SEPARATOR$PATH /bin/sh configure $configure_options --help) >/dev/null 2>&1; then
-  AT_CHECK([cd builddir && PATH=`pwd`/..$PATH_SEPARATOR$PATH /bin/sh configure $configure_options --help=recursive | grep " INNERMOST "], 0, [ignore])
+if (cd builddir && PATH=`pwd`/..$PATH_SEPARATOR$PATH @GENTOO_PORTAGE_EPREFIX@/bin/sh configure $configure_options --help) >/dev/null 2>&1; then
+  AT_CHECK([cd builddir && PATH=`pwd`/..$PATH_SEPARATOR$PATH @GENTOO_PORTAGE_EPREFIX@/bin/sh configure $configure_options --help=recursive | grep " INNERMOST "], 0, [ignore])
 fi
 AT_CHECK([PATH=`pwd`$PATH_SEPARATOR$PATH; export PATH; cd builddir && { configure $configure_options --help=recursive; chmod u+w .; } | grep " INNERMOST "],
 	 0, [ignore], [stderr],
@@ -1712,7 +1712,7 @@
 AS_MKDIR_P([inner])
 
 AT_DATA([inner/configure],
-[[#! /bin/sh
+[[#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 case "$*" in
     *--help*) echo 'No Autoconf here, folks!' ;;
     *)        echo got_it >myfile ;;
@@ -1762,7 +1762,7 @@
 AS_MKDIR_P([foo])
 
 AT_DATA([foo/configure],
-[[#! /bin/sh
+[[#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 touch innerfile
 exit 0
 ]])
