diff -urN a/cpan/Sys-Syslog/Makefile.PL b/cpan/Sys-Syslog/Makefile.PL
--- a/cpan/Sys-Syslog/Makefile.PL	2010-12-20 12:34:39.000000000 -0800
+++ b/cpan/Sys-Syslog/Makefile.PL	2012-02-12 08:18:40.502496800 -0800
@@ -1,7 +1,7 @@
 use strict;
 use Config;
 use ExtUtils::MakeMaker;
-eval 'use ExtUtils::MakeMaker::Coverage';
+#eval 'use ExtUtils::MakeMaker::Coverage';
 use File::Copy;
 use File::Path;
 use File::Spec;
