diff -urN Python-2.7.3.orig/Lib/test/test_posix.py Python-2.7.3/Lib/test/test_posix.py
--- Python-2.7.3.orig/Lib/test/test_posix.py	2012-04-09 16:07:32.000000000 -0700
+++ Python-2.7.3/Lib/test/test_posix.py	2012-10-06 03:06:21.284373700 -0700
@@ -107,7 +107,11 @@
         # If a non-privileged user invokes it, it should fail with OSError
         # EPERM.
         if os.getuid() != 0:
-            name = pwd.getpwuid(posix.getuid()).pw_name
+            try:
+                name = pwd.getpwuid(posix.getuid()).pw_name
+            except KeyError:
+                # the current UID may not have a pwd entry
+                raise unittest.SkipTest("need a pwd entry")
             try:
                 posix.initgroups(name, 13)
             except OSError as e:
@@ -412,8 +416,9 @@
     def test_getgroups(self):
         with os.popen('id -G') as idg:
             groups = idg.read().strip()
+            ret = idg.close()
 
-        if not groups:
+        if ret != None or not groups:
             raise unittest.SkipTest("need working 'id -G'")
 
         # 'id -G' and 'os.getgroups()' should return the same
