diff -urN ncurses-5.7.orig/misc/Makefile.in ncurses-5.7/misc/Makefile.in
--- ncurses-5.7.orig/misc/Makefile.in	2012-02-18 16:08:01.982373900 -0800
+++ ncurses-5.7/misc/Makefile.in	2012-02-18 16:14:42.764514300 -0800
@@ -107,7 +107,6 @@
 @MAKE_PC_FILES@install.libs :: pc-files $(DESTDIR)$(PKG_CONFIG_LIBDIR)
 @MAKE_PC_FILES@	$(SHELL) -c 'for name in *.pc; do $(INSTALL_DATA) $$name $(DESTDIR)$(PKG_CONFIG_LIBDIR)/$$name; done'
 
-@MAKE_PC_FILES@sources :: pc-files
 @MAKE_PC_FILES@pc-files :
 @MAKE_PC_FILES@	$(SHELL) ./gen-pkgconfig
 @MAKE_PC_FILES@	touch $@
