diff -u a/config.guess b/config.guess
--- a/config.guess	2012-02-23 15:13:07.079015400 -0800
+++ b/config.guess	2012-02-23 15:13:09.756015400 -0800
@@ -824,6 +824,9 @@
 		echo ${UNAME_MACHINE}-unknown-freebsd`echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'` ;;
 	esac
 	exit ;;
+    i*:CYGWIN*:1.7*:*)
+	echo ${UNAME_MACHINE}-pc-cygwin1.7
+	exit ;;
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
@@ -864,6 +867,10 @@
     i*:UWIN*:*)
 	echo ${UNAME_MACHINE}-pc-uwin
 	exit ;;
+
+    amd64:CYGWIN*:*:* | x86_64:CYGWIN*:1.7*:*)
+	echo x86_64-unknown-cygwin1.7
+	exit ;;
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
