--- gcc-4.5.3-r1.ebuild	2012-02-22 10:28:28.153436100 -0800
+++ /g2pfx/usr/portage/sys-devel/gcc/gcc-4.5.3-r1.ebuild	2011-11-30 11:01:33.000000000 -0800
@@ -17,7 +17,7 @@
 SSP_UCLIBC_STABLE=""
 #end Hardened stuff
 
-inherit toolchain flag-o-matic autotools
+inherit toolchain flag-o-matic
 
 DESCRIPTION="The GNU Compiler Collection."
 
@@ -71,94 +71,10 @@
 	PDEPEND="${PDEPEND} !prefix? ( elibc_glibc? ( >=sys-libs/glibc-2.8 ) )"
 fi
 
-#fixacreq() {
-#	acver="${1}"
-#	einfo "hacking AC_PREREQ in \"$(pwd)/configure.ac\""
-#	[[ $acver == 2.?? ]] || die "couldn't detect autoconf version"
-#	[[ -e configure.ac ]]  || die "no configure.ac to fix in \"$(pwd)\""
-#	sed -e 's/^AC_PREREQ(2\.[[:digit:]][[:digit:]])/AC_PREREQ('"${acver}"')/' -i configure.ac 
-#}
-#
-#fixm4acreq() {
-#	fil="${1}"
-#	acver="${2}"
-#	einfo "hacking m4 autoconf requirement macros in \"${fil}\""
-#	[[ $acver == 2.?? ]] || die "couldn't detect autoconf version"
-#	[[ -e $fil ]] || die "couldn't find file \"${fil}\""
-#	sed -e 's/m4_define(\[_GCC_AUTOCONF_VERSION], \[2\.[[:digit:]][[:digit:]]]/m4_define([_GCC_AUTOCONF_VERSION], ['"${acver}"']/' -i "${fil}"
-#	sed -e 's/m4_defn(\[m4_PACKAGE_VERSION]), \[2\.[[:digit:]][[:digit:]]]/m4_defn([m4_PACKAGE_VERSION]), ['"${acver}"']/' -i "${fil}"
-#}
-#
-## hack: we hook into exclude_gcc_patches only because it gives us the opportunity
-# to get our autotools updates in before toolchain.eclass starts messing around with
-# configure and other generated files
-#exclude_gcc_patches() {
-#	local i
-#	for i in ${GENTOO_PATCH_EXCLUDE} ; do
-#		if [[ -f ${WORKDIR}/patch/${i} ]] ; then
-#			einfo "Excluding patch ${i}"
-#			rm -f "${WORKDIR}"/patch/${i} || die "failed to delete ${i}"
-#		fi
-#	done
-#
-#	use vanilla && return 0
-#
-#	# cygwin hacks
-#	if [[ ${CHOST} == *-cygwin* ]] ; then
-#		pushd "${S}" || die
-#		epatch "${FILESDIR}"/gcc-4.5-cygwin-ada.patch
-#		epatch "${FILESDIR}"/gcc-4.5-cygwin-ehdebug.patch
-#		epatch "${FILESDIR}"/gcc-4.5.3-cygport.patch
-#		einfo "forcing some autotools crap for cygwin"
-#		local acver="$( autoconf --version| head -n 1|sed 's/^.*\(2.[[:digit:]][[:digit:]]\)/\1/' )"
-#		fixacreq $acver
-#		fixm4acreq "config/override.m4" $acver
-#		eautoconf || die 
-#		cd ${S}/gcc || die
-#		fixacreq $acver
-#		eautoconf || die
-#		eautoheader || die
-#		# problematic and pointless (no cygport changes anyhow)
-#		# cd ${S}/libiberty || di
-#		# fixacreq $acver
-#		# eautoconf || die
-#		cd ${S}/libjava || die
-#		fixacreq $acver
-#		eautoconf -f || die
-#		cd ${S}/libffi || die
-#		eaclocal -I . -I .. -I ../config || die
-#		fixacreq $acver
-#		eautoconf || die
-#		cd ${S} || die
-#		for x in boehm-gc libffi libgfortran libgomp libjava libmudflap libssp libstdc++-v3 zlib;
-#		do
-#			pushd $x || die
-#			einfo "( $(pwd) )"
-#			eautomake || die
-#			popd
-#		done
-#		cd ${S}/gcc/testsuite/ada/acats || die
-#		chmod a+x run_test.exp || die
-#		popd || die
-#		einfo 'yowza!  its done.'
-#	fi
-# }
-
 src_unpack() {
-
 	toolchain_src_unpack
 	use vanilla && return 0
 
-	epatch "${FILESDIR}"/gcc-4.5.3-hardpaths.patch
-	for f in \
-		libjava/classpath/examples/gnu/classpath/examples/awt/HintingDemo.java \
-		libjava/classpath/resource/gnu/java/awt/font/fonts.properties \
-		; do
-		einfo "Prefixifying hard-coded paths in \"${f}\""
-		[[ -e "${f}" ]] || { ewarn "oops... \"${f}\" not found !"; continue ; }
-		sed -e 's|@EPREFIX@|'"${EPREFIX%/}"'|' -i "${f}"
-	done
-
 	# work around http://gcc.gnu.org/bugzilla/show_bug.cgi?id=33637
 	epatch "${FILESDIR}"/4.3.0/targettools-checks.patch
 
@@ -202,7 +118,6 @@
 	sed -i 's/use_fixproto=yes/:/' gcc/config.gcc #PR33200
 
 	[[ ${CHOST} == ${CTARGET} ]] && epatch "${FILESDIR}"/gcc-spec-env.patch
-
 }
 
 pkg_setup() {
