diff -urN python-updater-0.10.orig/python-updater python-updater-0.10/python-updater
--- python-updater-0.10.orig/python-updater	2012-08-30 20:10:18.486381100 -0700
+++ python-updater-0.10/python-updater	2012-08-30 20:18:13.241740500 -0700
@@ -145,7 +145,8 @@
 # has_version(cpv)
 has_version() {
 	if [[ "${PMS_COMMAND[${PMS_INDEX}]}" == "emerge" ]]; then
-		"@GENTOO_PORTAGE_EPREFIX@"/usr/bin/portageq has_version / "${1}"
+		"@GENTOO_PORTAGE_EPREFIX@"/usr/bin/portageq has_version \
+                    "@GENTOO_PORTAGE_EPREFIX@"/ "${1}"
 	elif [[ "${PMS_COMMAND[${PMS_INDEX}]}" == "pmerge" ]]; then
 		pquery --vdb "${1}"
 	else
@@ -164,7 +165,7 @@
 
 import portage
 
-dbapi = portage.db['/']['porttree'].dbapi
+dbapi = portage.db['@GENTOO_PORTAGE_EPREFIX@/']['porttree'].dbapi
 pkg_list = dbapi.match('${1}/${2}${3:+:}${3}')
 if pkg_list:
 	best_visible = portage.best(pkg_list)
@@ -599,7 +600,7 @@
 for variable in variables:
 	exec('%(variable)s_regex = re.compile(r\'^%(variable)s=\((.*)\)\')' % {'variable': variable})
 
-dbapi = portage.db['/']['porttree'].dbapi
+dbapi = portage.db['@GENTOO_PORTAGE_EPREFIX@/']['porttree'].dbapi
 python_eclass_locations = {}
 
 if hasattr(dbapi.repositories.mainRepo(), 'eclass_db'):
